// dgz-ui-shared/dist/components/form/index.es.js
import { t as b } from "../../MyInput-BCm8-No1.js";
import { M as ie } from "../../MyInput-BCm8-No1.js";
import { M as me } from "../../MySelect-CDOMHlry.js";
import { j as e } from "../../jsx-runtime-C5mzlN2N.js";
import { l as v } from "../../lodash-CYNxjS-I.js";
import { v as h, k as u, N as g, I as j, R as p, T as o, X as k, P as C, q as M, a as S, E as w, G as F } from "../../input-mask-IfACg7qz-KwMI6P_O.js";
import { b as je, y as he, Q as ue, U as ge, c as ye, w as Ne, p as pe, x as fe, m as ve, g as be, B as ke, M as Ce, d as Me, f as Se, _ as we, e as Fe, h as Ee } from "../../input-mask-IfACg7qz-KwMI6P_O.js";
import { O as E, _ as I, u as R } from "../../popover-HTqpqYpJ-CIljcfgV.js";
import { u as P } from "../../button-Bp2lHjov-BTlHPcva.js";
import { d as G, n as z, L as _, D as $, G as D } from "../../DateRangePicker-A5Ubl8er.js";
import { m as y } from "../../utils-B6fNqzRf-B1_jG1K7.js";
import { c as q } from "../../createLucideIcon-B950nf2d.js";
import "react";
import { z as Re } from "../../react-select-Dr-07nIb-DE_aB_nz.js";
/**
 * @license lucide-react v0.511.0 - ISC
 *
 * This source code is licensed under the ISC license.
 * See the LICENSE file in the root directory of this source tree.
 */
const H = [
  ["path", { d: "M8 2v4", key: "1cmpym" }],
  ["path", { d: "M16 2v4", key: "4m81vk" }],
  ["rect", { width: "18", height: "18", x: "3", y: "4", rx: "2", key: "1hopcy" }],
  ["path", { d: "M3 10h18", key: "8toen8" }]
], L = q("calendar", H), Y = ({
  control: r,
  name: a,
  label: s,
  helperText: d,
  required: t,
  rules: c,
  format: n = z,
  placeholder: l,
  floatingError: i,
  register: m,
  ...x
}) => {
  const N = s && /* @__PURE__ */ e.jsxs(o, { className: "text-body-xs-medium my-3", children: [
    s,
    " ",
    t && /* @__PURE__ */ e.jsx("span", { className: "text-red-600", children: "*" })
  ] });
  return /* @__PURE__ */ e.jsx(e.Fragment, { children: a && r ? /* @__PURE__ */ e.jsx(
    h,
    {
      control: r,
      name: a,
      rules: c,
      render: ({ field: f, formState: B }) => /* @__PURE__ */ e.jsxs(u, { className: "relative flex flex-col", children: [
        N,
        /* @__PURE__ */ e.jsxs(E, { children: [
          /* @__PURE__ */ e.jsx(I, { asChild: !0, children: /* @__PURE__ */ e.jsx(g, { children: /* @__PURE__ */ e.jsxs(
            P,
            {
              variant: "secondary",
              className: y(
                "text-secondary !text-body-sm-regular border-alpha-strong focus:ring-item-primary mb-0 bg-transparent pl-3 text-left font-normal hover:bg-transparent",
                v.get(B.errors, `${a}`) && "focus:ring-item-destructive border-item-destructive bg-item-destructive-focus text-item-destructive hover:bg-item-destructive-focus dark:bg-transparent",
                f.value && "text-primary"
              ),
              children: [
                f.value ? G(f.value).format(n) : l || "Pick a date",
                /* @__PURE__ */ e.jsx(L, { className: "ml-auto h-4 w-4" })
              ]
            }
          ) }) }),
          /* @__PURE__ */ e.jsx(R, { className: "w-auto p-0", align: "start", children: /* @__PURE__ */ e.jsx(
            _,
            {
              ...x,
              mode: "single",
              selected: f.value,
              onSelect: f.onChange
            }
          ) })
        ] }),
        /* @__PURE__ */ e.jsx(j, { children: d }),
        /* @__PURE__ */ e.jsx("div", { className: "relative", children: /* @__PURE__ */ e.jsx(p, {}) })
      ] })
    }
  ) : /* @__PURE__ */ e.jsxs("div", { className: "relative flex flex-col", children: [
    N,
    /* @__PURE__ */ e.jsxs(E, { children: [
      /* @__PURE__ */ e.jsx(I, { asChild: !0, children: /* @__PURE__ */ e.jsxs(
        P,
        {
          variant: "secondary",
          className: y(
            "text-secondary !text-body-sm-regular border-border-alpha-strong focus:ring-item-primary mb-0 bg-transparent pl-3 text-left font-normal hover:bg-transparent",
            x.selected && "text-primary"
          ),
          children: [
            x.selected ? G(x.selected).format(n) : l || "Pick a date",
            /* @__PURE__ */ e.jsx(L, { className: "ml-auto h-4 w-4" })
          ]
        }
      ) }),
      /* @__PURE__ */ e.jsx(R, { className: "w-auto p-0", align: "start", children: /* @__PURE__ */ e.jsx(_, { ...x, mode: "single" }) })
    ] })
  ] }) });
}, Z = ({
  control: r,
  name: a,
  label: s,
  helperText: d,
  required: t,
  rules: c,
  format: n = z,
  placeholder: l,
  ...i
}) => {
  const m = s && /* @__PURE__ */ e.jsxs(o, { className: "text-body-xs-medium my-3", children: [
    s,
    " ",
    t && /* @__PURE__ */ e.jsx("span", { className: "text-red-600", children: "*" })
  ] });
  return /* @__PURE__ */ e.jsx(e.Fragment, { children: a && r && /* @__PURE__ */ e.jsx(
    h,
    {
      control: r,
      name: a,
      rules: c,
      render: ({ field: x, formState: N }) => /* @__PURE__ */ e.jsxs(u, { className: "relative flex flex-col", children: [
        m,
        /* @__PURE__ */ e.jsx(g, { children: /* @__PURE__ */ e.jsx(
          $,
          {
            ...i,
            error: `${v.get(N.errors, a, "")}`,
            selected: x.value,
            onRangeSelected: x.onChange,
            format: n,
            placeholder: l
          }
        ) }),
        /* @__PURE__ */ e.jsx(j, { children: d }),
        /* @__PURE__ */ e.jsx("div", { className: "relative", children: /* @__PURE__ */ e.jsx(p, {}) })
      ] })
    }
  ) });
}, ee = ({
  control: r,
  name: a,
  label: s,
  rules: d,
  helperText: t,
  ...c
}) => a && r ? /* @__PURE__ */ e.jsx(
  h,
  {
    control: r,
    name: a,
    rules: d,
    render: ({ field: n }) => /* @__PURE__ */ e.jsxs(u, { className: "flex flex-row items-start space-x-3", children: [
      /* @__PURE__ */ e.jsx(g, { children: /* @__PURE__ */ e.jsx(
        k,
        {
          checked: n.value,
          onCheckedChange: n.onChange,
          ...c
        }
      ) }),
      /* @__PURE__ */ e.jsxs("div", { className: "space-y-1 leading-none", children: [
        s && /* @__PURE__ */ e.jsx(o, { children: s }),
        /* @__PURE__ */ e.jsx(j, { children: t })
      ] })
    ] })
  }
) : /* @__PURE__ */ e.jsxs("div", { className: "flex flex-row items-start space-y-0 space-x-3 rounded-md border p-4", children: [
  /* @__PURE__ */ e.jsx(k, { ...c }),
  /* @__PURE__ */ e.jsx("div", { className: "space-y-1 leading-none", children: s && /* @__PURE__ */ e.jsx(o, { children: s }) })
] }), se = ({
  control: r,
  name: a,
  label: s,
  helperText: d,
  rules: t,
  ...c
}) => a && r ? /* @__PURE__ */ e.jsx(
  h,
  {
    control: r,
    name: a,
    rules: t,
    render: ({ field: n }) => /* @__PURE__ */ e.jsxs(u, { className: "flex items-center gap-2", children: [
      /* @__PURE__ */ e.jsx(g, { children: /* @__PURE__ */ e.jsx(
        C,
        {
          className: "m-0",
          checked: n.value,
          onCheckedChange: n.onChange,
          ...c
        }
      ) }),
      /* @__PURE__ */ e.jsxs("div", { className: "mb-1 space-y-1 leading-none", children: [
        s && /* @__PURE__ */ e.jsx(
          o,
          {
            className: y("text-primary", c.disabled && "opacity-40"),
            children: s
          }
        ),
        /* @__PURE__ */ e.jsx(j, { children: d })
      ] })
    ] })
  }
) : /* @__PURE__ */ e.jsxs("div", { className: "flex flex-row items-center space-x-3", children: [
  /* @__PURE__ */ e.jsx(C, { ...c }),
  /* @__PURE__ */ e.jsx("div", { className: "mb-1 space-y-1 leading-none", children: s && /* @__PURE__ */ e.jsx(
    o,
    {
      className: y("text-primary", c.disabled && "opacity-40"),
      children: s
    }
  ) })
] }), ae = ({
  control: r,
  name: a,
  label: s,
  helperText: d,
  required: t,
  className: c,
  rules: n,
  floatingError: l,
  ...i
}) => {
  const m = s && /* @__PURE__ */ e.jsxs(o, { className: "my-3", children: [
    s,
    " ",
    t && /* @__PURE__ */ e.jsx("span", { className: "text-red-600", children: "*" })
  ] });
  return a && r ? /* @__PURE__ */ e.jsx(
    h,
    {
      control: r,
      name: a,
      rules: n,
      render: ({ field: x }) => /* @__PURE__ */ e.jsxs(u, { children: [
        m,
        /* @__PURE__ */ e.jsx(g, { children: /* @__PURE__ */ e.jsx(D, { className: "mt-1", ...x, ...i }) }),
        /* @__PURE__ */ e.jsx(j, { children: d }),
        /* @__PURE__ */ e.jsx(p, { className: y(l && "absolute") })
      ] })
    }
  ) : /* @__PURE__ */ e.jsxs(e.Fragment, { children: [
    m,
    /* @__PURE__ */ e.jsx(D, { className: "mt-1", ...i })
  ] });
}, re = ({
  control: r,
  name: a,
  label: s,
  helperText: d,
  required: t,
  rules: c,
  className: n,
  floatingError: l,
  ...i
}) => {
  const m = s && /* @__PURE__ */ e.jsxs(o, { className: "text-body-xs-medium my-3", children: [
    s,
    " ",
    t && /* @__PURE__ */ e.jsx("span", { className: "text-red-600", children: "*" })
  ] });
  return a && r ? /* @__PURE__ */ e.jsx(
    h,
    {
      control: r,
      name: a,
      rules: c,
      render: ({ field: x, formState: N }) => /* @__PURE__ */ e.jsxs(u, { children: [
        m,
        /* @__PURE__ */ e.jsx(g, { children: /* @__PURE__ */ e.jsx(
          M,
          {
            variant: v.get(N.errors, `${a}`) ? "failure" : "default",
            ...i,
            ...x,
            className: b(["mt-2", n])
          }
        ) }),
        /* @__PURE__ */ e.jsx(j, { children: d }),
        /* @__PURE__ */ e.jsx(p, { className: y(l && "absolute") })
      ] })
    }
  ) : /* @__PURE__ */ e.jsxs(e.Fragment, { children: [
    m,
    /* @__PURE__ */ e.jsx(M, { ...i, className: b(["mt-2", n]) })
  ] });
}, te = ({
  control: r,
  name: a,
  label: s,
  rules: d,
  helperText: t,
  value: c,
  ...n
}) => a && r ? /* @__PURE__ */ e.jsx(
  h,
  {
    control: r,
    name: a,
    rules: d,
    render: ({ field: l }) => /* @__PURE__ */ e.jsxs(u, { className: "flex flex-row items-start space-x-3", children: [
      /* @__PURE__ */ e.jsx(g, { children: /* @__PURE__ */ e.jsx(
        S,
        {
          value: c,
          checked: l.value === c,
          onClick: () => l.onChange(c),
          ...n
        }
      ) }),
      /* @__PURE__ */ e.jsxs("div", { className: "space-y-1 leading-none", children: [
        s && /* @__PURE__ */ e.jsx(o, { children: s }),
        /* @__PURE__ */ e.jsx(j, { children: t })
      ] })
    ] })
  }
) : /* @__PURE__ */ e.jsxs("div", { className: "flex flex-row items-start space-y-0 space-x-3 rounded-md border p-4", children: [
  /* @__PURE__ */ e.jsx(S, { value: c, ...n }),
  /* @__PURE__ */ e.jsx("div", { className: "space-y-1 leading-none", children: s && /* @__PURE__ */ e.jsx(o, { children: s }) })
] }), ce = ({
  control: r,
  name: a,
  label: s,
  rules: d,
  helperText: t,
  required: c,
  floatingError: n,
  ...l
}) => {
  const i = s && /* @__PURE__ */ e.jsxs(o, { className: "text-body-xs-medium my-3", children: [
    s,
    " ",
    c && /* @__PURE__ */ e.jsx("span", { className: "text-red-600", children: "*" })
  ] });
  return a && r ? /* @__PURE__ */ e.jsx(
    h,
    {
      control: r,
      name: a,
      rules: d,
      render: ({ field: m }) => /* @__PURE__ */ e.jsxs(u, { children: [
        i,
        /* @__PURE__ */ e.jsx(g, { children: /* @__PURE__ */ e.jsx(w, { ...m, ...l }) }),
        t && /* @__PURE__ */ e.jsx(j, { children: t }),
        /* @__PURE__ */ e.jsx(p, { className: y(n && "absolute") })
      ] })
    }
  ) : /* @__PURE__ */ e.jsxs(e.Fragment, { children: [
    i,
    /* @__PURE__ */ e.jsx(w, { ...l }),
    t && /* @__PURE__ */ e.jsx(j, { children: t })
  ] });
}, ne = ({
  control: r,
  name: a,
  label: s,
  rules: d,
  helperText: t,
  required: c,
  floatingError: n,
  ...l
}) => {
  const i = s && /* @__PURE__ */ e.jsxs(o, { className: "text-body-xs-medium my-3", children: [
    s,
    " ",
    c && /* @__PURE__ */ e.jsx("span", { className: "text-red-600", children: "*" })
  ] });
  return a && r ? /* @__PURE__ */ e.jsx(
    h,
    {
      control: r,
      name: a,
      rules: d,
      render: ({ field: m }) => /* @__PURE__ */ e.jsxs(u, { children: [
        i,
        /* @__PURE__ */ e.jsx(g, { children: /* @__PURE__ */ e.jsx(
          F,
          {
            thousandsSeparator: " ",
            lazy: !1,
            placeholderChar: "_",
            unmask: !0,
            ...m,
            ...l,
            onAccept: (x) => m.onChange(x)
          }
        ) }),
        t && /* @__PURE__ */ e.jsx(j, { children: t }),
        /* @__PURE__ */ e.jsx(p, { className: y(n && "absolute") })
      ] })
    }
  ) : /* @__PURE__ */ e.jsxs(e.Fragment, { children: [
    i,
    /* @__PURE__ */ e.jsx(
      F,
      {
        thousandsSeparator: " ",
        lazy: !1,
        placeholderChar: "_",
        unmask: !0,
        ...l
      }
    ),
    t && /* @__PURE__ */ e.jsx(j, { children: t })
  ] });
};
export {
  k as Checkbox,
  je as FilepondContainer,
  he as Form,
  g as FormControl,
  j as FormDescription,
  h as FormField,
  u as FormItem,
  o as FormLabel,
  p as FormMessage,
  w as HtmlEditor,
  ue as Input,
  ge as Label,
  F as MaskInput,
  ee as MyCheckbox,
  Y as MyDatePicker,
  Z as MyDateRangePicker,
  ce as MyHtmlEditor,
  ie as MyInput,
  ne as MyMaskInput,
  te as MyRadio,
  me as MySelect,
  se as MySwitch,
  re as MyTextarea,
  ae as MyTimePicker,
  ye as RadioGroup,
  S as RadioGroupItem,
  Re as ReactSelect,
  Ne as Select,
  pe as SelectContent,
  fe as SelectGroup,
  ve as SelectItem,
  be as SelectLabel,
  ke as SelectScrollDownButton,
  Ce as SelectScrollUpButton,
  Me as SelectSeparator,
  Se as SelectTrigger,
  we as SelectValue,
  C as Switch,
  M as Textarea,
  Fe as inputVariants,
  Ee as useFormField
};
