// dgz-ui/dist/alert-qA-cFjdf.js
import { j as a } from "./jsx-runtime-C5mzlN2N.js";
import * as s from "react";
import { c as n } from "./index-BhbquNdX.js";
import { c as i } from "./utils-B6fNqzRf.js";
const d = n(
  "relative w-full rounded-lg border p-4 [&>svg~*]:pl-7 [&>svg+div]:translate-y-[-3px] [&>svg]:absolute [&>svg]:left-4 [&>svg]:top-4 [&>svg]:text-foreground",
  {
    variants: {
      variant: {
        default: "bg-background text-foreground",
        warning: "border-warning/50 text-warning dark:border-warning [&>svg]:text-warning",
        destructive: "border-destructive/50 text-destructive dark:border-destructive [&>svg]:text-destructive"
      }
    },
    defaultVariants: {
      variant: "default"
    }
  }
), l = s.forwardRef(({ className: t, variant: r, ...e }, o) => /* @__PURE__ */ a.jsx(
  "div",
  {
    ref: o,
    role: "alert",
    className: i(d({ variant: r }), t),
    ...e
  }
));
l.displayName = "Alert";
const g = s.forwardRef(({ className: t, ...r }, e) => /* @__PURE__ */ a.jsx(
  "h5",
  {
    ref: e,
    className: i("mb-1 leading-none font-medium tracking-tight", t),
    ...r
  }
));
g.displayName = "AlertTitle";
const c = s.forwardRef(({ className: t, ...r }, e) => /* @__PURE__ */ a.jsx(
  "div",
  {
    ref: e,
    className: i("text-sm [&_p]:leading-relaxed", t),
    ...r
  }
));
c.displayName = "AlertDescription";
export {
  l as A,
  d as a,
  g as b,
  c
};
