// dgz-ui/dist/breadcrumb-WSS8o6Gb.js
import { j as e } from "./jsx-runtime-C5mzlN2N.js";
import * as m from "react";
import { r as t } from "./index-Br4MVOGV.js";
import { c as i } from "./utils-B6fNqzRf.js";
import { E as c } from "./ellipsis-DZr9Fhhp.js";
const l = m.forwardRef(({ ...a }, r) => /* @__PURE__ */ e.jsx("nav", { ref: r, "aria-label": "breadcrumb", ...a }));
l.displayName = "Breadcrumb";
const o = m.forwardRef(({ className: a, ...r }, s) => /* @__PURE__ */ e.jsx(
  "ol",
  {
    ref: s,
    className: i("flex flex-wrap items-center gap-1.5 break-words", a),
    ...r
  }
));
o.displayName = "BreadcrumbList";
const d = m.forwardRef(({ className: a, ...r }, s) => /* @__PURE__ */ e.jsx(
  "li",
  {
    ref: s,
    className: i("inline-flex items-center gap-1.5", a),
    ...r
  }
));
d.displayName = "BreadcrumbItem";
const n = m.forwardRef(({ className: a, ...r }, s) => /* @__PURE__ */ e.jsx(
  "span",
  {
    ref: s,
    role: "link",
    "aria-disabled": "true",
    "aria-current": "page",
    className: i("text-foreground font-normal", a),
    ...r
  }
));
n.displayName = "BreadcrumbPage";
const p = ({
  children: a,
  className: r,
  ...s
}) => /* @__PURE__ */ e.jsx(
  "li",
  {
    role: "presentation",
    "aria-hidden": "true",
    className: i("text-secondary [&>svg]:size-4", r),
    ...s,
    children: a ?? /* @__PURE__ */ e.jsx(t, { size: 20 })
  }
);
p.displayName = "BreadcrumbSeparator";
const f = ({
  className: a,
  ...r
}) => /* @__PURE__ */ e.jsxs(
  "span",
  {
    role: "presentation",
    "aria-hidden": "true",
    className: i("flex h-9 w-9 items-center justify-center", a),
    ...r,
    children: [
      /* @__PURE__ */ e.jsx(c, { className: "h-4 w-4" }),
      /* @__PURE__ */ e.jsx("span", { className: "sr-only", children: "More" })
    ]
  }
);
f.displayName = "BreadcrumbElipssis";
export {
  l as B,
  o as a,
  d as b,
  n as c,
  p as d,
  f as e
};
