// dgz-ui/dist/card-ZYjMXH_J.js
import { j as r } from "./jsx-runtime-C5mzlN2N.js";
import * as d from "react";
import { c as t } from "./utils-B6fNqzRf.js";
const o = d.forwardRef(
  ({ className: a, ...e }, s) => /* @__PURE__ */ r.jsx(
    "div",
    {
      ref: s,
      className: t(
        "rounded-4 border-border-alpha-light bg-bg-background border shadow-xs",
        a
      ),
      ...e
    }
  )
);
o.displayName = "Card";
const i = d.forwardRef(({ className: a, ...e }, s) => /* @__PURE__ */ r.jsx(
  "div",
  {
    ref: s,
    className: t("flex flex-col space-y-1.5 p-3", a),
    ...e
  }
));
i.displayName = "CardHeader";
const m = d.forwardRef(({ className: a, ...e }, s) => /* @__PURE__ */ r.jsx(
  "div",
  {
    ref: s,
    className: t(
      "text-body-md-medium flex items-center justify-between gap-2 [&>div>svg]:size-5 [&>svg]:size-5",
      a
    ),
    ...e
  }
));
m.displayName = "CardTitle";
const l = d.forwardRef(({ className: a, ...e }, s) => /* @__PURE__ */ r.jsx(
  "span",
  {
    ref: s,
    className: t("text-body-sm-regular text-secondary", a),
    ...e
  }
));
l.displayName = "CardDescription";
const c = d.forwardRef(({ className: a, ...e }, s) => /* @__PURE__ */ r.jsx(
  "div",
  {
    ref: s,
    className: t("text-body-sm-regular p-3 pt-0", a),
    ...e
  }
));
c.displayName = "CardContent";
const p = d.forwardRef(({ className: a, ...e }, s) => /* @__PURE__ */ r.jsx(
  "div",
  {
    ref: s,
    className: t("text-body-sm-regular flex items-center p-3 pt-0", a),
    ...e
  }
));
p.displayName = "CardFooter";
export {
  o as C,
  i as a,
  p as b,
  m as c,
  l as d,
  c as e
};
