// dgz-ui/dist/index.es.js
import { A as o, c as r, b as s, a as t } from "./alert-qA-cFjdf.js";
import { A as i, i as n, j as p, d as g, h as d, f as c, e as b, b as m, a as D, g as u, c as T } from "./alert-dialog-DO5x00uf.js";
import { A as C, c as f, b as x, a as A } from "./avatar-D2XIyeP7.js";
import { B as P, b as w } from "./badge-DGmo3xSE.js";
import { B as h, e as B, b as I, a as v, c as k, d as H } from "./breadcrumb-WSS8o6Gb.js";
import { B as L, b as E } from "./button-Bp2lHjov.js";
import { C as G, e as j, d as y, b as O, a as U, c as N } from "./card-ZYjMXH_J.js";
import { C as z, D as J, a as K, H as Q, M as W, T as X, b as Y } from "./timepicker-Cv7giYZn.js";
import { D as _, c as $, e as aa, i as ea, g as oa, f as ra, b as sa, a as ta, h as la, d as ia } from "./dialog-BWAsJVnp.js";
import { D as pa, d as ga, b as da, i as ca, c as ba, f as ma, j as Da, n as ua, e as Ta, g as Sa, h as Ca, k as fa, l as xa, m as Aa, a as Ma } from "./dropdown-menu-BPBcxHRB.js";
import { C as wa, F as Fa, a as ha, d as Ba, e as Ia, g as va, b as ka, c as Ha, f as Ra, H as La, I as Ea, L as Va, M as Ga, R as ja, s as ya, h as Oa, m as Ua, j as Na, o as qa, n as za, r as Ja, q as Ka, p as Qa, l as Wa, k as Xa, S as Ya, T as Za, i as _a, u as $a } from "./input-mask-IfACg7qz.js";
import { R as ee } from "./react-select-Dr-07nIb.js";
import { P as re, a as se, b as te, c as le, d as ie, e as ne, f as pe } from "./pagination-1SPhXT81.js";
import { P as de, b as ce, a as be } from "./popover-HTqpqYpJ.js";
import { S as De, a as ue } from "./scroll-area-JqD01-EB.js";
import { S as Se } from "./separator-Dno6nZNU.js";
import { S as fe, b as xe, c as Ae, g as Me, e as Pe, d as we, f as Fe, a as he } from "./sheet-psfPkdYr.js";
import { T as Ie, c as ve, a as ke, b as He } from "./tabs-DQAuE_IR.js";
import { T as Le, b as Ee, g as Ve, f as Ge, c as je, d as ye, a as Oe, e as Ue } from "./table-BE7fnN23.js";
import { T as qe, b as ze, c as Je, a as Ke } from "./tooltip-C35SqwxH.js";
import { S as We } from "./skeleton-BdurNoF7.js";
import { C as Ye, b as Ze, a as _e } from "./collapsible-DvKy678y.js";
import { c as ao } from "./utils-B6fNqzRf.js";
export {
  o as Alert,
  r as AlertDescription,
  i as AlertDialog,
  n as AlertDialogAction,
  p as AlertDialogCancel,
  g as AlertDialogContent,
  d as AlertDialogDescription,
  c as AlertDialogFooter,
  b as AlertDialogHeader,
  m as AlertDialogOverlay,
  D as AlertDialogPortal,
  u as AlertDialogTitle,
  T as AlertDialogTrigger,
  s as AlertTitle,
  C as Avatar,
  f as AvatarFallback,
  x as AvatarImage,
  P as Badge,
  h as Breadcrumb,
  B as BreadcrumbEllipsis,
  I as BreadcrumbItem,
  v as BreadcrumbList,
  k as BreadcrumbPage,
  H as BreadcrumbSeparator,
  L as Button,
  z as Calendar,
  G as Card,
  j as CardContent,
  y as CardDescription,
  O as CardFooter,
  U as CardHeader,
  N as CardTitle,
  wa as Checkbox,
  Ye as Collapsible,
  Ze as CollapsibleContent,
  _e as CollapsibleTrigger,
  J as DATE,
  K as DatePicker,
  _ as Dialog,
  $ as DialogClose,
  aa as DialogContent,
  ea as DialogDescription,
  oa as DialogFooter,
  ra as DialogHeader,
  sa as DialogOverlay,
  ta as DialogPortal,
  la as DialogTitle,
  ia as DialogTrigger,
  pa as DropdownMenu,
  ga as DropdownMenuCheckboxItem,
  da as DropdownMenuContent,
  ca as DropdownMenuGroup,
  ba as DropdownMenuItem,
  ma as DropdownMenuLabel,
  Da as DropdownMenuPortal,
  ua as DropdownMenuRadioGroup,
  Ta as DropdownMenuRadioItem,
  Sa as DropdownMenuSeparator,
  Ca as DropdownMenuShortcut,
  fa as DropdownMenuSub,
  xa as DropdownMenuSubContent,
  Aa as DropdownMenuSubTrigger,
  Ma as DropdownMenuTrigger,
  Fa as FilepondContainer,
  ha as Form,
  Ba as FormControl,
  Ia as FormDescription,
  va as FormField,
  ka as FormItem,
  Ha as FormLabel,
  Ra as FormMessage,
  Q as HOUR,
  La as HtmlEditor,
  Ea as Input,
  Va as Label,
  W as MINUTE,
  Ga as MaskInput,
  re as Pagination,
  se as PaginationContent,
  te as PaginationEllipsis,
  le as PaginationItem,
  ie as PaginationLink,
  ne as PaginationNext,
  pe as PaginationPrevious,
  de as Popover,
  ce as PopoverContent,
  be as PopoverTrigger,
  ja as RadioGroup,
  ya as RadioGroupItem,
  ee as ReactSelect,
  De as ScrollArea,
  ue as ScrollBar,
  Oa as Select,
  Ua as SelectContent,
  Na as SelectGroup,
  qa as SelectItem,
  za as SelectLabel,
  Ja as SelectScrollDownButton,
  Ka as SelectScrollUpButton,
  Qa as SelectSeparator,
  Wa as SelectTrigger,
  Xa as SelectValue,
  Se as Separator,
  fe as Sheet,
  xe as SheetClose,
  Ae as SheetContent,
  Me as SheetDescription,
  Pe as SheetFooter,
  we as SheetHeader,
  Fe as SheetTitle,
  he as SheetTrigger,
  We as Skeleton,
  Ya as Switch,
  X as TIME,
  Le as Table,
  Ee as TableBody,
  Ve as TableCaption,
  Ge as TableCell,
  je as TableFooter,
  ye as TableHead,
  Oe as TableHeader,
  Ue as TableRow,
  Ie as Tabs,
  ve as TabsContent,
  ke as TabsList,
  He as TabsTrigger,
  Za as Textarea,
  Y as TimePicker,
  qe as Tooltip,
  ze as TooltipContent,
  Je as TooltipProvider,
  Ke as TooltipTrigger,
  t as alertVariants,
  A as avatarVariants,
  w as badgeVariants,
  E as buttonVariants,
  ao as cn,
  _a as inputVariants,
  $a as useFormField
};
