// dgz-ui/dist/pagination-1SPhXT81.js
import { j as i } from "./jsx-runtime-C5mzlN2N.js";
import * as o from "react";
import { b as l } from "./button-Bp2lHjov.js";
import { c as e } from "./utils-B6fNqzRf.js";
import { E as m } from "./ellipsis-DZr9Fhhp.js";
import { C as p } from "./chevron-right-HWBijj-I.js";
import { C as c } from "./chevron-left-CQZFYxtY.js";
const g = ({ className: s, ...a }) => /* @__PURE__ */ i.jsx(
  "nav",
  {
    role: "navigation",
    "aria-label": "pagination",
    className: e("flex w-full justify-center", s),
    ...a
  }
);
g.displayName = "Pagination";
const x = o.forwardRef(({ className: s, ...a }, n) => /* @__PURE__ */ i.jsx(
  "ul",
  {
    ref: n,
    className: e("flex flex-row items-center gap-1", s),
    ...a
  }
));
x.displayName = "PaginationContent";
const f = o.forwardRef(({ className: s, ...a }, n) => /* @__PURE__ */ i.jsx("li", { ref: n, className: e("", s), ...a }));
f.displayName = "PaginationItem";
const t = ({
  className: s,
  isActive: a,
  size: n = "icon",
  ...r
}) => /* @__PURE__ */ i.jsx(
  "a",
  {
    "aria-current": a ? "page" : void 0,
    className: e(
      l({
        variant: a ? "tertiary" : "ghost",
        size: n
      }),
      "cursor-pointer",
      s
    ),
    ...r
  }
);
t.displayName = "PaginationLink";
const N = ({
  className: s,
  ...a
}) => /* @__PURE__ */ i.jsx(
  t,
  {
    "aria-label": "Go to previous page",
    size: "default",
    className: e("gap-1 px-2.5", s),
    ...a,
    children: /* @__PURE__ */ i.jsx(c, { className: "h-4 w-4" })
  }
);
N.displayName = "PaginationPrevious";
const d = ({
  className: s,
  ...a
}) => /* @__PURE__ */ i.jsx(
  t,
  {
    "aria-label": "Go to next page",
    size: "default",
    className: e("gap-1 px-2.5", s),
    ...a,
    children: /* @__PURE__ */ i.jsx(p, { className: "h-4 w-4" })
  }
);
d.displayName = "PaginationNext";
const P = ({
  className: s,
  ...a
}) => /* @__PURE__ */ i.jsxs(
  "span",
  {
    "aria-hidden": !0,
    className: e("flex h-9 w-9 items-center justify-center", s),
    ...a,
    children: [
      /* @__PURE__ */ i.jsx(m, { className: "h-4 w-4" }),
      /* @__PURE__ */ i.jsx("span", { className: "sr-only", children: "More pages" })
    ]
  }
);
P.displayName = "PaginationEllipsis";
export {
  g as P,
  x as a,
  P as b,
  f as c,
  t as d,
  d as e,
  N as f
};
