// dgz-ui-shared/dist/Loader-CVKJNLck.js
import { j as t } from "./jsx-runtime-C5mzlN2N.js";
import { m as s } from "./utils-B6fNqzRf-B1_jG1K7.js";
const a = (e) => /* @__PURE__ */ t.jsxs(
  "svg",
  {
    xmlns: "http://www.w3.org/2000/svg",
    fill: "none",
    viewBox: "0 0 24 24",
    ...e,
    className: s("mr-1 size-5 animate-spin text-white", e == null ? void 0 : e.className),
    children: [
      /* @__PURE__ */ t.jsx(
        "circle",
        {
          className: "opacity-25",
          cx: "12",
          cy: "12",
          r: "10",
          stroke: "currentColor",
          strokeWidth: "4"
        }
      ),
      /* @__PURE__ */ t.jsx(
        "path",
        {
          className: "opacity-75",
          fill: "currentColor",
          d: "M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4zm2 5.291A7.962 7.962 0 014 12H0c0 3.042 1.135 5.824 3 7.938l3-2.647z"
        }
      )
    ]
  }
), m = (e) => /* @__PURE__ */ t.jsx(
  "div",
  {
    className: "text-item-primary mx-auto flex w-full items-center justify-center p-4",
    ...e,
    children: /* @__PURE__ */ t.jsx(a, { className: "text-item-primary size-6" })
  }
);
export {
  m as L,
  a as S
};
