// dgz-ui-shared/dist/MySelect-CDOMHlry.js
import { j as s } from "./jsx-runtime-C5mzlN2N.js";
import { I as l, v as u, k as E, N as R, R as k, T as y } from "./input-mask-IfACg7qz-KwMI6P_O.js";
import { z as d } from "./react-select-Dr-07nIb-DE_aB_nz.js";
import { l as z } from "./lodash-CYNxjS-I.js";
import { m as h } from "./utils-B6fNqzRf-B1_jG1K7.js";
const S = ({
  control: e,
  name: r,
  label: a,
  helperText: m,
  required: g,
  className: t,
  rules: f,
  options: n = [],
  onChange: o,
  ...x
}) => {
  const c = a && /* @__PURE__ */ s.jsxs(y, { className: "my-3", children: [
    a,
    " ",
    g && /* @__PURE__ */ s.jsx("span", { className: "text-red-600", children: "*" })
  ] });
  return !r || !e ? /* @__PURE__ */ s.jsxs(s.Fragment, { children: [
    c,
    /* @__PURE__ */ s.jsx(
      d,
      {
        className: h("mt-2", t),
        ...x,
        options: n
      }
    ),
    /* @__PURE__ */ s.jsx(l, { children: m })
  ] }) : /* @__PURE__ */ s.jsx(
    u,
    {
      control: e,
      name: r,
      rules: f,
      render: ({ field: i, formState: p }) => {
        const N = (j) => {
          i.onChange(j), o && o(j);
        };
        return /* @__PURE__ */ s.jsxs(E, { children: [
          c,
          /* @__PURE__ */ s.jsx(R, { children: /* @__PURE__ */ s.jsx(
            d,
            {
              className: h("mt-2", t),
              ...x,
              ...i,
              onChange: N,
              options: n,
              error: !!z.get(p.errors, `${r}`)
            }
          ) }),
          /* @__PURE__ */ s.jsx(l, { children: m }),
          /* @__PURE__ */ s.jsx(k, {})
        ] });
      }
    }
  );
};
export {
  S as M
};
