// dgz-ui-shared/dist/unstyled-DA_k1WVz.js
import { j as R } from "./jsx-runtime-C5mzlN2N.js";
import * as P from "react";
import j, { useMemo as ls, useState as Ze, useRef as he, useLayoutEffect as kt, useEffect as Ee, useSyncExternalStore as ds, isValidElement as Ae, cloneElement as qe } from "react";
import { u as bt } from "./useTranslation-dH-uwymc.js";
import { a as O, x as wt, j as cs, b as us, _ as fs } from "./index-DqTIeHVj-CeLMF2B2.js";
import { D as Tt, v as hs, x as ms, C as ps, m as gs, _ as ys, h as vs, P as Ct, E as _s, R as xs } from "./index-sde7W6lA-CGLFJGTK.js";
import { m as te } from "./utils-B6fNqzRf-B1_jG1K7.js";
import { g as Nt } from "./button-Bp2lHjov-BTlHPcva.js";
import { y as ks } from "./input-mask-IfACg7qz-KwMI6P_O.js";
import { M as bs } from "./MyInput-BCm8-No1.js";
import { g as Le, s as Se, a as ws, u as Ts } from "./index.esm-lbl2HgR1.js";
var Et = "AlertDialog", [Cs, ia] = cs(Et, [
  Tt
]), X = Tt(), At = (t) => {
  const { __scopeAlertDialog: e, ...s } = t, r = X(e);
  return /* @__PURE__ */ O.jsx(hs, { ...r, ...s, modal: !0 });
};
At.displayName = Et;
var Ns = "AlertDialogTrigger", It = P.forwardRef(
  (t, e) => {
    const { __scopeAlertDialog: s, ...r } = t, a = X(s);
    return /* @__PURE__ */ O.jsx(ms, { ...a, ...r, ref: e });
  }
);
It.displayName = Ns;
var Es = "AlertDialogPortal", jt = (t) => {
  const { __scopeAlertDialog: e, ...s } = t, r = X(e);
  return /* @__PURE__ */ O.jsx(ps, { ...r, ...s });
};
jt.displayName = Es;
var As = "AlertDialogOverlay", Ot = P.forwardRef(
  (t, e) => {
    const { __scopeAlertDialog: s, ...r } = t, a = X(s);
    return /* @__PURE__ */ O.jsx(xs, { ...a, ...r, ref: e });
  }
);
Ot.displayName = As;
var oe = "AlertDialogContent", [Is, js] = Cs(oe), Os = fs("AlertDialogContent"), Rt = P.forwardRef(
  (t, e) => {
    const { __scopeAlertDialog: s, children: r, ...a } = t, n = X(s), i = P.useRef(null), o = wt(e, i), l = P.useRef(null);
    return /* @__PURE__ */ O.jsx(
      gs,
      {
        contentName: oe,
        titleName: St,
        docsSlug: "alert-dialog",
        children: /* @__PURE__ */ O.jsx(Is, { scope: s, cancelRef: l, children: /* @__PURE__ */ O.jsxs(
          ys,
          {
            role: "alertdialog",
            ...n,
            ...a,
            ref: o,
            onOpenAutoFocus: us(a.onOpenAutoFocus, (d) => {
              var f;
              d.preventDefault(), (f = l.current) == null || f.focus({ preventScroll: !0 });
            }),
            onPointerDownOutside: (d) => d.preventDefault(),
            onInteractOutside: (d) => d.preventDefault(),
            children: [
              /* @__PURE__ */ O.jsx(Os, { children: r }),
              /* @__PURE__ */ O.jsx(Ss, { contentRef: i })
            ]
          }
        ) })
      }
    );
  }
);
Rt.displayName = oe;
var St = "AlertDialogTitle", $t = P.forwardRef(
  (t, e) => {
    const { __scopeAlertDialog: s, ...r } = t, a = X(s);
    return /* @__PURE__ */ O.jsx(vs, { ...a, ...r, ref: e });
  }
);
$t.displayName = St;
var Zt = "AlertDialogDescription", Lt = P.forwardRef((t, e) => {
  const { __scopeAlertDialog: s, ...r } = t, a = X(s);
  return /* @__PURE__ */ O.jsx(_s, { ...a, ...r, ref: e });
});
Lt.displayName = Zt;
var Rs = "AlertDialogAction", Pt = P.forwardRef(
  (t, e) => {
    const { __scopeAlertDialog: s, ...r } = t, a = X(s);
    return /* @__PURE__ */ O.jsx(Ct, { ...a, ...r, ref: e });
  }
);
Pt.displayName = Rs;
var Dt = "AlertDialogCancel", Mt = P.forwardRef(
  (t, e) => {
    const { __scopeAlertDialog: s, ...r } = t, { cancelRef: a } = js(Dt, s), n = X(s), i = wt(e, a);
    return /* @__PURE__ */ O.jsx(Ct, { ...n, ...r, ref: i });
  }
);
Mt.displayName = Dt;
var Ss = ({ contentRef: t }) => {
  const e = `\`${oe}\` requires a description for the component to be accessible for screen reader users.

You can add a description to the \`${oe}\` by passing a \`${Zt}\` component as a child, which also benefits sighted users by adding visible context to the dialog.

Alternatively, you can use your own component as a description by assigning it an \`id\` and passing the same value to the \`aria-describedby\` prop in \`${oe}\`. If the description is confusing or duplicative for sighted users, you can use the \`@radix-ui/react-visually-hidden\` primitive as a wrapper around your description component.

For more information, see https://radix-ui.com/primitives/docs/components/alert-dialog`;
  return P.useEffect(() => {
    var s;
    document.getElementById(
      (s = t.current) == null ? void 0 : s.getAttribute("aria-describedby")
    ) || console.warn(e);
  }, [e, t]), null;
}, $s = At, Zs = It, Ls = jt, zt = Ot, Vt = Rt, Bt = Pt, Ft = Mt, Ut = $t, Wt = Lt;
const qt = $s, Ht = Zs, Ps = Ls, Xt = P.forwardRef(({ className: t, ...e }, s) => /* @__PURE__ */ O.jsx(
  zt,
  {
    className: te(
      "data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 fixed inset-0 z-50 bg-black/80",
      t
    ),
    ...e,
    ref: s
  }
));
Xt.displayName = zt.displayName;
const He = P.forwardRef(({ className: t, ...e }, s) => /* @__PURE__ */ O.jsxs(Ps, { children: [
  /* @__PURE__ */ O.jsx(Xt, {}),
  /* @__PURE__ */ O.jsx(
    Vt,
    {
      ref: s,
      className: te(
        "bg-background data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[state=closed]:slide-out-to-left-1/2 data-[state=closed]:slide-out-to-top-[48%] data-[state=open]:slide-in-from-left-1/2 data-[state=open]:slide-in-from-top-[48%] fixed top-[50%] left-[50%] z-50 grid w-full max-w-lg translate-x-[-50%] translate-y-[-50%] gap-4 border p-6 shadow-lg duration-200 sm:rounded-lg",
        t
      ),
      ...e
    }
  )
] }));
He.displayName = Vt.displayName;
const Xe = ({
  className: t,
  ...e
}) => /* @__PURE__ */ O.jsx(
  "div",
  {
    className: te(
      "flex flex-col space-y-2 text-center sm:text-left",
      t
    ),
    ...e
  }
);
Xe.displayName = "AlertDialogHeader";
const Ye = ({
  className: t,
  ...e
}) => /* @__PURE__ */ O.jsx("div", { className: te("flex gap-2 sm:justify-end", t), ...e });
Ye.displayName = "AlertDialogFooter";
const Je = P.forwardRef(({ className: t, ...e }, s) => /* @__PURE__ */ O.jsx(
  Ut,
  {
    ref: s,
    className: te("text-lg font-semibold", t),
    ...e
  }
));
Je.displayName = Ut.displayName;
const Yt = P.forwardRef(({ className: t, ...e }, s) => /* @__PURE__ */ O.jsx(
  Wt,
  {
    ref: s,
    className: te("text-muted-foreground text-sm", t),
    ...e
  }
));
Yt.displayName = Wt.displayName;
const Qe = P.forwardRef(({ className: t, ...e }, s) => /* @__PURE__ */ O.jsx(
  Bt,
  {
    ref: s,
    className: te(Nt(), t),
    ...e
  }
));
Qe.displayName = Bt.displayName;
const Ge = P.forwardRef(({ className: t, ...e }, s) => /* @__PURE__ */ O.jsx(
  Ft,
  {
    ref: s,
    className: te(Nt({ variant: "tertiary" }), t),
    ...e
  }
));
Ge.displayName = Ft.displayName;
const oa = ({
  title: t,
  description: e,
  children: s,
  onConfirm: r,
  ...a
}) => {
  const { t: n } = bt();
  return /* @__PURE__ */ R.jsxs(qt, { ...a, children: [
    s && /* @__PURE__ */ R.jsx(Ht, { children: s }),
    /* @__PURE__ */ R.jsxs(He, { children: [
      /* @__PURE__ */ R.jsxs(Xe, { children: [
        /* @__PURE__ */ R.jsx(Je, { children: t || n("Are you absolutely sure?") }),
        /* @__PURE__ */ R.jsx(Yt, { children: e || n(
          "This action cannot be undone. This will permanently delete your data from our servers."
        ) })
      ] }),
      /* @__PURE__ */ R.jsxs(Ye, { children: [
        /* @__PURE__ */ R.jsx(Ge, { className: "mt-0", children: n("Cancel") }),
        /* @__PURE__ */ R.jsx(Qe, { onClick: r, children: n("Confirm") })
      ] })
    ] })
  ] });
}, st = (t, e, s) => {
  if (t && "reportValidity" in t) {
    const r = Le(s, e);
    t.setCustomValidity(r && r.message || ""), t.reportValidity();
  }
}, Jt = (t, e) => {
  for (const s in e.fields) {
    const r = e.fields[s];
    r && r.ref && "reportValidity" in r.ref ? st(r.ref, s, t) : r && r.refs && r.refs.forEach((a) => st(a, s, t));
  }
}, Ds = (t, e) => {
  e.shouldUseNativeValidation && Jt(t, e);
  const s = {};
  for (const r in t) {
    const a = Le(e.fields, r), n = Object.assign(t[r] || {}, { ref: a && a.ref });
    if (Ms(e.names || Object.keys(t), r)) {
      const i = Object.assign({}, Le(s, r));
      Se(i, "root", n), Se(s, r, i);
    } else Se(s, r, n);
  }
  return s;
}, Ms = (t, e) => {
  const s = rt(e);
  return t.some((r) => rt(r).match(`^${s}\\.\\d+`));
};
function rt(t) {
  return t.replace(/\]|\[/g, "");
}
function zs(t, e) {
  for (var s = {}; t.length; ) {
    var r = t[0], a = r.code, n = r.message, i = r.path.join(".");
    if (!s[i]) if ("unionErrors" in r) {
      var o = r.unionErrors[0].errors[0];
      s[i] = { message: o.message, type: o.code };
    } else s[i] = { message: n, type: a };
    if ("unionErrors" in r && r.unionErrors.forEach(function(f) {
      return f.errors.forEach(function(E) {
        return t.push(E);
      });
    }), e) {
      var l = s[i].types, d = l && l[r.code];
      s[i] = ws(i, e, s, a, d ? [].concat(d, r.message) : r.message);
    }
    t.shift();
  }
  return s;
}
function Vs(t, e, s) {
  return s === void 0 && (s = {}), function(r, a, n) {
    try {
      return Promise.resolve(function(i, o) {
        try {
          var l = Promise.resolve(t[s.mode === "sync" ? "parse" : "parseAsync"](r, e)).then(function(d) {
            return n.shouldUseNativeValidation && Jt({}, n), { errors: {}, values: s.raw ? Object.assign({}, r) : d };
          });
        } catch (d) {
          return o(d);
        }
        return l && l.then ? l.then(void 0, o) : l;
      }(0, function(i) {
        if (function(o) {
          return Array.isArray(o == null ? void 0 : o.errors);
        }(i)) return { values: {}, errors: Ds(zs(i.errors, !n.shouldUseNativeValidation && n.criteriaMode === "all"), n) };
        throw i;
      }));
    } catch (i) {
      return Promise.reject(i);
    }
  };
}
var b;
(function(t) {
  t.assertEqual = (a) => {
  };
  function e(a) {
  }
  t.assertIs = e;
  function s(a) {
    throw new Error();
  }
  t.assertNever = s, t.arrayToEnum = (a) => {
    const n = {};
    for (const i of a)
      n[i] = i;
    return n;
  }, t.getValidEnumValues = (a) => {
    const n = t.objectKeys(a).filter((o) => typeof a[a[o]] != "number"), i = {};
    for (const o of n)
      i[o] = a[o];
    return t.objectValues(i);
  }, t.objectValues = (a) => t.objectKeys(a).map(function(n) {
    return a[n];
  }), t.objectKeys = typeof Object.keys == "function" ? (a) => Object.keys(a) : (a) => {
    const n = [];
    for (const i in a)
      Object.prototype.hasOwnProperty.call(a, i) && n.push(i);
    return n;
  }, t.find = (a, n) => {
    for (const i of a)
      if (n(i))
        return i;
  }, t.isInteger = typeof Number.isInteger == "function" ? (a) => Number.isInteger(a) : (a) => typeof a == "number" && Number.isFinite(a) && Math.floor(a) === a;
  function r(a, n = " | ") {
    return a.map((i) => typeof i == "string" ? `'${i}'` : i).join(n);
  }
  t.joinValues = r, t.jsonStringifyReplacer = (a, n) => typeof n == "bigint" ? n.toString() : n;
})(b || (b = {}));
var at;
(function(t) {
  t.mergeShapes = (e, s) => ({
    ...e,
    ...s
    // second overwrites first
  });
})(at || (at = {}));
const h = b.arrayToEnum([
  "string",
  "nan",
  "number",
  "integer",
  "float",
  "boolean",
  "date",
  "bigint",
  "symbol",
  "function",
  "undefined",
  "null",
  "array",
  "object",
  "unknown",
  "promise",
  "void",
  "never",
  "map",
  "set"
]), J = (t) => {
  switch (typeof t) {
    case "undefined":
      return h.undefined;
    case "string":
      return h.string;
    case "number":
      return Number.isNaN(t) ? h.nan : h.number;
    case "boolean":
      return h.boolean;
    case "function":
      return h.function;
    case "bigint":
      return h.bigint;
    case "symbol":
      return h.symbol;
    case "object":
      return Array.isArray(t) ? h.array : t === null ? h.null : t.then && typeof t.then == "function" && t.catch && typeof t.catch == "function" ? h.promise : typeof Map < "u" && t instanceof Map ? h.map : typeof Set < "u" && t instanceof Set ? h.set : typeof Date < "u" && t instanceof Date ? h.date : h.object;
    default:
      return h.unknown;
  }
}, c = b.arrayToEnum([
  "invalid_type",
  "invalid_literal",
  "custom",
  "invalid_union",
  "invalid_union_discriminator",
  "invalid_enum_value",
  "unrecognized_keys",
  "invalid_arguments",
  "invalid_return_type",
  "invalid_date",
  "invalid_string",
  "too_small",
  "too_big",
  "invalid_intersection_types",
  "not_multiple_of",
  "not_finite"
]);
class q extends Error {
  get errors() {
    return this.issues;
  }
  constructor(e) {
    super(), this.issues = [], this.addIssue = (r) => {
      this.issues = [...this.issues, r];
    }, this.addIssues = (r = []) => {
      this.issues = [...this.issues, ...r];
    };
    const s = new.target.prototype;
    Object.setPrototypeOf ? Object.setPrototypeOf(this, s) : this.__proto__ = s, this.name = "ZodError", this.issues = e;
  }
  format(e) {
    const s = e || function(n) {
      return n.message;
    }, r = { _errors: [] }, a = (n) => {
      for (const i of n.issues)
        if (i.code === "invalid_union")
          i.unionErrors.map(a);
        else if (i.code === "invalid_return_type")
          a(i.returnTypeError);
        else if (i.code === "invalid_arguments")
          a(i.argumentsError);
        else if (i.path.length === 0)
          r._errors.push(s(i));
        else {
          let o = r, l = 0;
          for (; l < i.path.length; ) {
            const d = i.path[l];
            l === i.path.length - 1 ? (o[d] = o[d] || { _errors: [] }, o[d]._errors.push(s(i))) : o[d] = o[d] || { _errors: [] }, o = o[d], l++;
          }
        }
    };
    return a(this), r;
  }
  static assert(e) {
    if (!(e instanceof q))
      throw new Error(`Not a ZodError: ${e}`);
  }
  toString() {
    return this.message;
  }
  get message() {
    return JSON.stringify(this.issues, b.jsonStringifyReplacer, 2);
  }
  get isEmpty() {
    return this.issues.length === 0;
  }
  flatten(e = (s) => s.message) {
    const s = {}, r = [];
    for (const a of this.issues)
      a.path.length > 0 ? (s[a.path[0]] = s[a.path[0]] || [], s[a.path[0]].push(e(a))) : r.push(e(a));
    return { formErrors: r, fieldErrors: s };
  }
  get formErrors() {
    return this.flatten();
  }
}
q.create = (t) => new q(t);
const Pe = (t, e) => {
  let s;
  switch (t.code) {
    case c.invalid_type:
      t.received === h.undefined ? s = "Required" : s = `Expected ${t.expected}, received ${t.received}`;
      break;
    case c.invalid_literal:
      s = `Invalid literal value, expected ${JSON.stringify(t.expected, b.jsonStringifyReplacer)}`;
      break;
    case c.unrecognized_keys:
      s = `Unrecognized key(s) in object: ${b.joinValues(t.keys, ", ")}`;
      break;
    case c.invalid_union:
      s = "Invalid input";
      break;
    case c.invalid_union_discriminator:
      s = `Invalid discriminator value. Expected ${b.joinValues(t.options)}`;
      break;
    case c.invalid_enum_value:
      s = `Invalid enum value. Expected ${b.joinValues(t.options)}, received '${t.received}'`;
      break;
    case c.invalid_arguments:
      s = "Invalid function arguments";
      break;
    case c.invalid_return_type:
      s = "Invalid function return type";
      break;
    case c.invalid_date:
      s = "Invalid date";
      break;
    case c.invalid_string:
      typeof t.validation == "object" ? "includes" in t.validation ? (s = `Invalid input: must include "${t.validation.includes}"`, typeof t.validation.position == "number" && (s = `${s} at one or more positions greater than or equal to ${t.validation.position}`)) : "startsWith" in t.validation ? s = `Invalid input: must start with "${t.validation.startsWith}"` : "endsWith" in t.validation ? s = `Invalid input: must end with "${t.validation.endsWith}"` : b.assertNever(t.validation) : t.validation !== "regex" ? s = `Invalid ${t.validation}` : s = "Invalid";
      break;
    case c.too_small:
      t.type === "array" ? s = `Array must contain ${t.exact ? "exactly" : t.inclusive ? "at least" : "more than"} ${t.minimum} element(s)` : t.type === "string" ? s = `String must contain ${t.exact ? "exactly" : t.inclusive ? "at least" : "over"} ${t.minimum} character(s)` : t.type === "number" ? s = `Number must be ${t.exact ? "exactly equal to " : t.inclusive ? "greater than or equal to " : "greater than "}${t.minimum}` : t.type === "date" ? s = `Date must be ${t.exact ? "exactly equal to " : t.inclusive ? "greater than or equal to " : "greater than "}${new Date(Number(t.minimum))}` : s = "Invalid input";
      break;
    case c.too_big:
      t.type === "array" ? s = `Array must contain ${t.exact ? "exactly" : t.inclusive ? "at most" : "less than"} ${t.maximum} element(s)` : t.type === "string" ? s = `String must contain ${t.exact ? "exactly" : t.inclusive ? "at most" : "under"} ${t.maximum} character(s)` : t.type === "number" ? s = `Number must be ${t.exact ? "exactly" : t.inclusive ? "less than or equal to" : "less than"} ${t.maximum}` : t.type === "bigint" ? s = `BigInt must be ${t.exact ? "exactly" : t.inclusive ? "less than or equal to" : "less than"} ${t.maximum}` : t.type === "date" ? s = `Date must be ${t.exact ? "exactly" : t.inclusive ? "smaller than or equal to" : "smaller than"} ${new Date(Number(t.maximum))}` : s = "Invalid input";
      break;
    case c.custom:
      s = "Invalid input";
      break;
    case c.invalid_intersection_types:
      s = "Intersection results could not be merged";
      break;
    case c.not_multiple_of:
      s = `Number must be a multiple of ${t.multipleOf}`;
      break;
    case c.not_finite:
      s = "Number must be finite";
      break;
    default:
      s = e.defaultError, b.assertNever(t);
  }
  return { message: s };
};
let Bs = Pe;
function Fs() {
  return Bs;
}
const Us = (t) => {
  const { data: e, path: s, errorMaps: r, issueData: a } = t, n = [...s, ...a.path || []], i = {
    ...a,
    path: n
  };
  if (a.message !== void 0)
    return {
      ...a,
      path: n,
      message: a.message
    };
  let o = "";
  const l = r.filter((d) => !!d).slice().reverse();
  for (const d of l)
    o = d(i, { data: e, defaultError: o }).message;
  return {
    ...a,
    path: n,
    message: o
  };
};
function u(t, e) {
  const s = Fs(), r = Us({
    issueData: e,
    data: t.data,
    path: t.path,
    errorMaps: [
      t.common.contextualErrorMap,
      // contextual error map is first priority
      t.schemaErrorMap,
      // then schema-bound map if available
      s,
      // then global override map
      s === Pe ? void 0 : Pe
      // then global default map
    ].filter((a) => !!a)
  });
  t.common.issues.push(r);
}
class V {
  constructor() {
    this.value = "valid";
  }
  dirty() {
    this.value === "valid" && (this.value = "dirty");
  }
  abort() {
    this.value !== "aborted" && (this.value = "aborted");
  }
  static mergeArray(e, s) {
    const r = [];
    for (const a of s) {
      if (a.status === "aborted")
        return y;
      a.status === "dirty" && e.dirty(), r.push(a.value);
    }
    return { status: e.value, value: r };
  }
  static async mergeObjectAsync(e, s) {
    const r = [];
    for (const a of s) {
      const n = await a.key, i = await a.value;
      r.push({
        key: n,
        value: i
      });
    }
    return V.mergeObjectSync(e, r);
  }
  static mergeObjectSync(e, s) {
    const r = {};
    for (const a of s) {
      const { key: n, value: i } = a;
      if (n.status === "aborted" || i.status === "aborted")
        return y;
      n.status === "dirty" && e.dirty(), i.status === "dirty" && e.dirty(), n.value !== "__proto__" && (typeof i.value < "u" || a.alwaysSet) && (r[n.value] = i.value);
    }
    return { status: e.value, value: r };
  }
}
const y = Object.freeze({
  status: "aborted"
}), fe = (t) => ({ status: "dirty", value: t }), B = (t) => ({ status: "valid", value: t }), nt = (t) => t.status === "aborted", it = (t) => t.status === "dirty", le = (t) => t.status === "valid", be = (t) => typeof Promise < "u" && t instanceof Promise;
var m;
(function(t) {
  t.errToObj = (e) => typeof e == "string" ? { message: e } : e || {}, t.toString = (e) => typeof e == "string" ? e : e == null ? void 0 : e.message;
})(m || (m = {}));
class K {
  constructor(e, s, r, a) {
    this._cachedPath = [], this.parent = e, this.data = s, this._path = r, this._key = a;
  }
  get path() {
    return this._cachedPath.length || (Array.isArray(this._key) ? this._cachedPath.push(...this._path, ...this._key) : this._cachedPath.push(...this._path, this._key)), this._cachedPath;
  }
}
const ot = (t, e) => {
  if (le(e))
    return { success: !0, data: e.value };
  if (!t.common.issues.length)
    throw new Error("Validation failed but no issues detected.");
  return {
    success: !1,
    get error() {
      if (this._error)
        return this._error;
      const s = new q(t.common.issues);
      return this._error = s, this._error;
    }
  };
};
function _(t) {
  if (!t)
    return {};
  const { errorMap: e, invalid_type_error: s, required_error: r, description: a } = t;
  if (e && (s || r))
    throw new Error(`Can't use "invalid_type_error" or "required_error" in conjunction with custom error map.`);
  return e ? { errorMap: e, description: a } : { errorMap: (i, o) => {
    const { message: l } = t;
    return i.code === "invalid_enum_value" ? { message: l ?? o.defaultError } : typeof o.data > "u" ? { message: l ?? r ?? o.defaultError } : i.code !== "invalid_type" ? { message: o.defaultError } : { message: l ?? s ?? o.defaultError };
  }, description: a };
}
class k {
  get description() {
    return this._def.description;
  }
  _getType(e) {
    return J(e.data);
  }
  _getOrReturnCtx(e, s) {
    return s || {
      common: e.parent.common,
      data: e.data,
      parsedType: J(e.data),
      schemaErrorMap: this._def.errorMap,
      path: e.path,
      parent: e.parent
    };
  }
  _processInputParams(e) {
    return {
      status: new V(),
      ctx: {
        common: e.parent.common,
        data: e.data,
        parsedType: J(e.data),
        schemaErrorMap: this._def.errorMap,
        path: e.path,
        parent: e.parent
      }
    };
  }
  _parseSync(e) {
    const s = this._parse(e);
    if (be(s))
      throw new Error("Synchronous parse encountered promise.");
    return s;
  }
  _parseAsync(e) {
    const s = this._parse(e);
    return Promise.resolve(s);
  }
  parse(e, s) {
    const r = this.safeParse(e, s);
    if (r.success)
      return r.data;
    throw r.error;
  }
  safeParse(e, s) {
    const r = {
      common: {
        issues: [],
        async: (s == null ? void 0 : s.async) ?? !1,
        contextualErrorMap: s == null ? void 0 : s.errorMap
      },
      path: (s == null ? void 0 : s.path) || [],
      schemaErrorMap: this._def.errorMap,
      parent: null,
      data: e,
      parsedType: J(e)
    }, a = this._parseSync({ data: e, path: r.path, parent: r });
    return ot(r, a);
  }
  "~validate"(e) {
    var r, a;
    const s = {
      common: {
        issues: [],
        async: !!this["~standard"].async
      },
      path: [],
      schemaErrorMap: this._def.errorMap,
      parent: null,
      data: e,
      parsedType: J(e)
    };
    if (!this["~standard"].async)
      try {
        const n = this._parseSync({ data: e, path: [], parent: s });
        return le(n) ? {
          value: n.value
        } : {
          issues: s.common.issues
        };
      } catch (n) {
        (a = (r = n == null ? void 0 : n.message) == null ? void 0 : r.toLowerCase()) != null && a.includes("encountered") && (this["~standard"].async = !0), s.common = {
          issues: [],
          async: !0
        };
      }
    return this._parseAsync({ data: e, path: [], parent: s }).then((n) => le(n) ? {
      value: n.value
    } : {
      issues: s.common.issues
    });
  }
  async parseAsync(e, s) {
    const r = await this.safeParseAsync(e, s);
    if (r.success)
      return r.data;
    throw r.error;
  }
  async safeParseAsync(e, s) {
    const r = {
      common: {
        issues: [],
        contextualErrorMap: s == null ? void 0 : s.errorMap,
        async: !0
      },
      path: (s == null ? void 0 : s.path) || [],
      schemaErrorMap: this._def.errorMap,
      parent: null,
      data: e,
      parsedType: J(e)
    }, a = this._parse({ data: e, path: r.path, parent: r }), n = await (be(a) ? a : Promise.resolve(a));
    return ot(r, n);
  }
  refine(e, s) {
    const r = (a) => typeof s == "string" || typeof s > "u" ? { message: s } : typeof s == "function" ? s(a) : s;
    return this._refinement((a, n) => {
      const i = e(a), o = () => n.addIssue({
        code: c.custom,
        ...r(a)
      });
      return typeof Promise < "u" && i instanceof Promise ? i.then((l) => l ? !0 : (o(), !1)) : i ? !0 : (o(), !1);
    });
  }
  refinement(e, s) {
    return this._refinement((r, a) => e(r) ? !0 : (a.addIssue(typeof s == "function" ? s(r, a) : s), !1));
  }
  _refinement(e) {
    return new ce({
      schema: this,
      typeName: v.ZodEffects,
      effect: { type: "refinement", refinement: e }
    });
  }
  superRefine(e) {
    return this._refinement(e);
  }
  constructor(e) {
    this.spa = this.safeParseAsync, this._def = e, this.parse = this.parse.bind(this), this.safeParse = this.safeParse.bind(this), this.parseAsync = this.parseAsync.bind(this), this.safeParseAsync = this.safeParseAsync.bind(this), this.spa = this.spa.bind(this), this.refine = this.refine.bind(this), this.refinement = this.refinement.bind(this), this.superRefine = this.superRefine.bind(this), this.optional = this.optional.bind(this), this.nullable = this.nullable.bind(this), this.nullish = this.nullish.bind(this), this.array = this.array.bind(this), this.promise = this.promise.bind(this), this.or = this.or.bind(this), this.and = this.and.bind(this), this.transform = this.transform.bind(this), this.brand = this.brand.bind(this), this.default = this.default.bind(this), this.catch = this.catch.bind(this), this.describe = this.describe.bind(this), this.pipe = this.pipe.bind(this), this.readonly = this.readonly.bind(this), this.isNullable = this.isNullable.bind(this), this.isOptional = this.isOptional.bind(this), this["~standard"] = {
      version: 1,
      vendor: "zod",
      validate: (s) => this["~validate"](s)
    };
  }
  optional() {
    return G.create(this, this._def);
  }
  nullable() {
    return ue.create(this, this._def);
  }
  nullish() {
    return this.nullable().optional();
  }
  array() {
    return U.create(this);
  }
  promise() {
    return Ne.create(this, this._def);
  }
  or(e) {
    return Te.create([this, e], this._def);
  }
  and(e) {
    return Ce.create(this, e, this._def);
  }
  transform(e) {
    return new ce({
      ..._(this._def),
      schema: this,
      typeName: v.ZodEffects,
      effect: { type: "transform", transform: e }
    });
  }
  default(e) {
    const s = typeof e == "function" ? e : () => e;
    return new Me({
      ..._(this._def),
      innerType: this,
      defaultValue: s,
      typeName: v.ZodDefault
    });
  }
  brand() {
    return new hr({
      typeName: v.ZodBranded,
      type: this,
      ..._(this._def)
    });
  }
  catch(e) {
    const s = typeof e == "function" ? e : () => e;
    return new ze({
      ..._(this._def),
      innerType: this,
      catchValue: s,
      typeName: v.ZodCatch
    });
  }
  describe(e) {
    const s = this.constructor;
    return new s({
      ...this._def,
      description: e
    });
  }
  pipe(e) {
    return Ke.create(this, e);
  }
  readonly() {
    return Ve.create(this);
  }
  isOptional() {
    return this.safeParse(void 0).success;
  }
  isNullable() {
    return this.safeParse(null).success;
  }
}
const Ws = /^c[^\s-]{8,}$/i, qs = /^[0-9a-z]+$/, Hs = /^[0-9A-HJKMNP-TV-Z]{26}$/i, Xs = /^[0-9a-fA-F]{8}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{12}$/i, Ys = /^[a-z0-9_-]{21}$/i, Js = /^[A-Za-z0-9-_]+\.[A-Za-z0-9-_]+\.[A-Za-z0-9-_]*$/, Qs = /^[-+]?P(?!$)(?:(?:[-+]?\d+Y)|(?:[-+]?\d+[.,]\d+Y$))?(?:(?:[-+]?\d+M)|(?:[-+]?\d+[.,]\d+M$))?(?:(?:[-+]?\d+W)|(?:[-+]?\d+[.,]\d+W$))?(?:(?:[-+]?\d+D)|(?:[-+]?\d+[.,]\d+D$))?(?:T(?=[\d+-])(?:(?:[-+]?\d+H)|(?:[-+]?\d+[.,]\d+H$))?(?:(?:[-+]?\d+M)|(?:[-+]?\d+[.,]\d+M$))?(?:[-+]?\d+(?:[.,]\d+)?S)?)??$/, Gs = /^(?!\.)(?!.*\.\.)([A-Z0-9_'+\-\.]*)[A-Z0-9_+-]@([A-Z0-9][A-Z0-9\-]*\.)+[A-Z]{2,}$/i, Ks = "^(\\p{Extended_Pictographic}|\\p{Emoji_Component})+$";
let $e;
const er = /^(?:(?:25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9][0-9]|[0-9])\.){3}(?:25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9][0-9]|[0-9])$/, tr = /^(?:(?:25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9][0-9]|[0-9])\.){3}(?:25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9][0-9]|[0-9])\/(3[0-2]|[12]?[0-9])$/, sr = /^(([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,7}:|([0-9a-fA-F]{1,4}:){1,6}:[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,5}(:[0-9a-fA-F]{1,4}){1,2}|([0-9a-fA-F]{1,4}:){1,4}(:[0-9a-fA-F]{1,4}){1,3}|([0-9a-fA-F]{1,4}:){1,3}(:[0-9a-fA-F]{1,4}){1,4}|([0-9a-fA-F]{1,4}:){1,2}(:[0-9a-fA-F]{1,4}){1,5}|[0-9a-fA-F]{1,4}:((:[0-9a-fA-F]{1,4}){1,6})|:((:[0-9a-fA-F]{1,4}){1,7}|:)|fe80:(:[0-9a-fA-F]{0,4}){0,4}%[0-9a-zA-Z]{1,}|::(ffff(:0{1,4}){0,1}:){0,1}((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])|([0-9a-fA-F]{1,4}:){1,4}:((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9]))$/, rr = /^(([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,7}:|([0-9a-fA-F]{1,4}:){1,6}:[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,5}(:[0-9a-fA-F]{1,4}){1,2}|([0-9a-fA-F]{1,4}:){1,4}(:[0-9a-fA-F]{1,4}){1,3}|([0-9a-fA-F]{1,4}:){1,3}(:[0-9a-fA-F]{1,4}){1,4}|([0-9a-fA-F]{1,4}:){1,2}(:[0-9a-fA-F]{1,4}){1,5}|[0-9a-fA-F]{1,4}:((:[0-9a-fA-F]{1,4}){1,6})|:((:[0-9a-fA-F]{1,4}){1,7}|:)|fe80:(:[0-9a-fA-F]{0,4}){0,4}%[0-9a-zA-Z]{1,}|::(ffff(:0{1,4}){0,1}:){0,1}((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])|([0-9a-fA-F]{1,4}:){1,4}:((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9]))\/(12[0-8]|1[01][0-9]|[1-9]?[0-9])$/, ar = /^([0-9a-zA-Z+/]{4})*(([0-9a-zA-Z+/]{2}==)|([0-9a-zA-Z+/]{3}=))?$/, nr = /^([0-9a-zA-Z-_]{4})*(([0-9a-zA-Z-_]{2}(==)?)|([0-9a-zA-Z-_]{3}(=)?))?$/, Qt = "((\\d\\d[2468][048]|\\d\\d[13579][26]|\\d\\d0[48]|[02468][048]00|[13579][26]00)-02-29|\\d{4}-((0[13578]|1[02])-(0[1-9]|[12]\\d|3[01])|(0[469]|11)-(0[1-9]|[12]\\d|30)|(02)-(0[1-9]|1\\d|2[0-8])))", ir = new RegExp(`^${Qt}$`);
function Gt(t) {
  let e = "[0-5]\\d";
  t.precision ? e = `${e}\\.\\d{${t.precision}}` : t.precision == null && (e = `${e}(\\.\\d+)?`);
  const s = t.precision ? "+" : "?";
  return `([01]\\d|2[0-3]):[0-5]\\d(:${e})${s}`;
}
function or(t) {
  return new RegExp(`^${Gt(t)}$`);
}
function lr(t) {
  let e = `${Qt}T${Gt(t)}`;
  const s = [];
  return s.push(t.local ? "Z?" : "Z"), t.offset && s.push("([+-]\\d{2}:?\\d{2})"), e = `${e}(${s.join("|")})`, new RegExp(`^${e}$`);
}
function dr(t, e) {
  return !!((e === "v4" || !e) && er.test(t) || (e === "v6" || !e) && sr.test(t));
}
function cr(t, e) {
  if (!Js.test(t))
    return !1;
  try {
    const [s] = t.split("."), r = s.replace(/-/g, "+").replace(/_/g, "/").padEnd(s.length + (4 - s.length % 4) % 4, "="), a = JSON.parse(atob(r));
    return !(typeof a != "object" || a === null || "typ" in a && (a == null ? void 0 : a.typ) !== "JWT" || !a.alg || e && a.alg !== e);
  } catch {
    return !1;
  }
}
function ur(t, e) {
  return !!((e === "v4" || !e) && tr.test(t) || (e === "v6" || !e) && rr.test(t));
}
class Q extends k {
  _parse(e) {
    if (this._def.coerce && (e.data = String(e.data)), this._getType(e) !== h.string) {
      const n = this._getOrReturnCtx(e);
      return u(n, {
        code: c.invalid_type,
        expected: h.string,
        received: n.parsedType
      }), y;
    }
    const r = new V();
    let a;
    for (const n of this._def.checks)
      if (n.kind === "min")
        e.data.length < n.value && (a = this._getOrReturnCtx(e, a), u(a, {
          code: c.too_small,
          minimum: n.value,
          type: "string",
          inclusive: !0,
          exact: !1,
          message: n.message
        }), r.dirty());
      else if (n.kind === "max")
        e.data.length > n.value && (a = this._getOrReturnCtx(e, a), u(a, {
          code: c.too_big,
          maximum: n.value,
          type: "string",
          inclusive: !0,
          exact: !1,
          message: n.message
        }), r.dirty());
      else if (n.kind === "length") {
        const i = e.data.length > n.value, o = e.data.length < n.value;
        (i || o) && (a = this._getOrReturnCtx(e, a), i ? u(a, {
          code: c.too_big,
          maximum: n.value,
          type: "string",
          inclusive: !0,
          exact: !0,
          message: n.message
        }) : o && u(a, {
          code: c.too_small,
          minimum: n.value,
          type: "string",
          inclusive: !0,
          exact: !0,
          message: n.message
        }), r.dirty());
      } else if (n.kind === "email")
        Gs.test(e.data) || (a = this._getOrReturnCtx(e, a), u(a, {
          validation: "email",
          code: c.invalid_string,
          message: n.message
        }), r.dirty());
      else if (n.kind === "emoji")
        $e || ($e = new RegExp(Ks, "u")), $e.test(e.data) || (a = this._getOrReturnCtx(e, a), u(a, {
          validation: "emoji",
          code: c.invalid_string,
          message: n.message
        }), r.dirty());
      else if (n.kind === "uuid")
        Xs.test(e.data) || (a = this._getOrReturnCtx(e, a), u(a, {
          validation: "uuid",
          code: c.invalid_string,
          message: n.message
        }), r.dirty());
      else if (n.kind === "nanoid")
        Ys.test(e.data) || (a = this._getOrReturnCtx(e, a), u(a, {
          validation: "nanoid",
          code: c.invalid_string,
          message: n.message
        }), r.dirty());
      else if (n.kind === "cuid")
        Ws.test(e.data) || (a = this._getOrReturnCtx(e, a), u(a, {
          validation: "cuid",
          code: c.invalid_string,
          message: n.message
        }), r.dirty());
      else if (n.kind === "cuid2")
        qs.test(e.data) || (a = this._getOrReturnCtx(e, a), u(a, {
          validation: "cuid2",
          code: c.invalid_string,
          message: n.message
        }), r.dirty());
      else if (n.kind === "ulid")
        Hs.test(e.data) || (a = this._getOrReturnCtx(e, a), u(a, {
          validation: "ulid",
          code: c.invalid_string,
          message: n.message
        }), r.dirty());
      else if (n.kind === "url")
        try {
          new URL(e.data);
        } catch {
          a = this._getOrReturnCtx(e, a), u(a, {
            validation: "url",
            code: c.invalid_string,
            message: n.message
          }), r.dirty();
        }
      else n.kind === "regex" ? (n.regex.lastIndex = 0, n.regex.test(e.data) || (a = this._getOrReturnCtx(e, a), u(a, {
        validation: "regex",
        code: c.invalid_string,
        message: n.message
      }), r.dirty())) : n.kind === "trim" ? e.data = e.data.trim() : n.kind === "includes" ? e.data.includes(n.value, n.position) || (a = this._getOrReturnCtx(e, a), u(a, {
        code: c.invalid_string,
        validation: { includes: n.value, position: n.position },
        message: n.message
      }), r.dirty()) : n.kind === "toLowerCase" ? e.data = e.data.toLowerCase() : n.kind === "toUpperCase" ? e.data = e.data.toUpperCase() : n.kind === "startsWith" ? e.data.startsWith(n.value) || (a = this._getOrReturnCtx(e, a), u(a, {
        code: c.invalid_string,
        validation: { startsWith: n.value },
        message: n.message
      }), r.dirty()) : n.kind === "endsWith" ? e.data.endsWith(n.value) || (a = this._getOrReturnCtx(e, a), u(a, {
        code: c.invalid_string,
        validation: { endsWith: n.value },
        message: n.message
      }), r.dirty()) : n.kind === "datetime" ? lr(n).test(e.data) || (a = this._getOrReturnCtx(e, a), u(a, {
        code: c.invalid_string,
        validation: "datetime",
        message: n.message
      }), r.dirty()) : n.kind === "date" ? ir.test(e.data) || (a = this._getOrReturnCtx(e, a), u(a, {
        code: c.invalid_string,
        validation: "date",
        message: n.message
      }), r.dirty()) : n.kind === "time" ? or(n).test(e.data) || (a = this._getOrReturnCtx(e, a), u(a, {
        code: c.invalid_string,
        validation: "time",
        message: n.message
      }), r.dirty()) : n.kind === "duration" ? Qs.test(e.data) || (a = this._getOrReturnCtx(e, a), u(a, {
        validation: "duration",
        code: c.invalid_string,
        message: n.message
      }), r.dirty()) : n.kind === "ip" ? dr(e.data, n.version) || (a = this._getOrReturnCtx(e, a), u(a, {
        validation: "ip",
        code: c.invalid_string,
        message: n.message
      }), r.dirty()) : n.kind === "jwt" ? cr(e.data, n.alg) || (a = this._getOrReturnCtx(e, a), u(a, {
        validation: "jwt",
        code: c.invalid_string,
        message: n.message
      }), r.dirty()) : n.kind === "cidr" ? ur(e.data, n.version) || (a = this._getOrReturnCtx(e, a), u(a, {
        validation: "cidr",
        code: c.invalid_string,
        message: n.message
      }), r.dirty()) : n.kind === "base64" ? ar.test(e.data) || (a = this._getOrReturnCtx(e, a), u(a, {
        validation: "base64",
        code: c.invalid_string,
        message: n.message
      }), r.dirty()) : n.kind === "base64url" ? nr.test(e.data) || (a = this._getOrReturnCtx(e, a), u(a, {
        validation: "base64url",
        code: c.invalid_string,
        message: n.message
      }), r.dirty()) : b.assertNever(n);
    return { status: r.value, value: e.data };
  }
  _regex(e, s, r) {
    return this.refinement((a) => e.test(a), {
      validation: s,
      code: c.invalid_string,
      ...m.errToObj(r)
    });
  }
  _addCheck(e) {
    return new Q({
      ...this._def,
      checks: [...this._def.checks, e]
    });
  }
  email(e) {
    return this._addCheck({ kind: "email", ...m.errToObj(e) });
  }
  url(e) {
    return this._addCheck({ kind: "url", ...m.errToObj(e) });
  }
  emoji(e) {
    return this._addCheck({ kind: "emoji", ...m.errToObj(e) });
  }
  uuid(e) {
    return this._addCheck({ kind: "uuid", ...m.errToObj(e) });
  }
  nanoid(e) {
    return this._addCheck({ kind: "nanoid", ...m.errToObj(e) });
  }
  cuid(e) {
    return this._addCheck({ kind: "cuid", ...m.errToObj(e) });
  }
  cuid2(e) {
    return this._addCheck({ kind: "cuid2", ...m.errToObj(e) });
  }
  ulid(e) {
    return this._addCheck({ kind: "ulid", ...m.errToObj(e) });
  }
  base64(e) {
    return this._addCheck({ kind: "base64", ...m.errToObj(e) });
  }
  base64url(e) {
    return this._addCheck({
      kind: "base64url",
      ...m.errToObj(e)
    });
  }
  jwt(e) {
    return this._addCheck({ kind: "jwt", ...m.errToObj(e) });
  }
  ip(e) {
    return this._addCheck({ kind: "ip", ...m.errToObj(e) });
  }
  cidr(e) {
    return this._addCheck({ kind: "cidr", ...m.errToObj(e) });
  }
  datetime(e) {
    return typeof e == "string" ? this._addCheck({
      kind: "datetime",
      precision: null,
      offset: !1,
      local: !1,
      message: e
    }) : this._addCheck({
      kind: "datetime",
      precision: typeof (e == null ? void 0 : e.precision) > "u" ? null : e == null ? void 0 : e.precision,
      offset: (e == null ? void 0 : e.offset) ?? !1,
      local: (e == null ? void 0 : e.local) ?? !1,
      ...m.errToObj(e == null ? void 0 : e.message)
    });
  }
  date(e) {
    return this._addCheck({ kind: "date", message: e });
  }
  time(e) {
    return typeof e == "string" ? this._addCheck({
      kind: "time",
      precision: null,
      message: e
    }) : this._addCheck({
      kind: "time",
      precision: typeof (e == null ? void 0 : e.precision) > "u" ? null : e == null ? void 0 : e.precision,
      ...m.errToObj(e == null ? void 0 : e.message)
    });
  }
  duration(e) {
    return this._addCheck({ kind: "duration", ...m.errToObj(e) });
  }
  regex(e, s) {
    return this._addCheck({
      kind: "regex",
      regex: e,
      ...m.errToObj(s)
    });
  }
  includes(e, s) {
    return this._addCheck({
      kind: "includes",
      value: e,
      position: s == null ? void 0 : s.position,
      ...m.errToObj(s == null ? void 0 : s.message)
    });
  }
  startsWith(e, s) {
    return this._addCheck({
      kind: "startsWith",
      value: e,
      ...m.errToObj(s)
    });
  }
  endsWith(e, s) {
    return this._addCheck({
      kind: "endsWith",
      value: e,
      ...m.errToObj(s)
    });
  }
  min(e, s) {
    return this._addCheck({
      kind: "min",
      value: e,
      ...m.errToObj(s)
    });
  }
  max(e, s) {
    return this._addCheck({
      kind: "max",
      value: e,
      ...m.errToObj(s)
    });
  }
  length(e, s) {
    return this._addCheck({
      kind: "length",
      value: e,
      ...m.errToObj(s)
    });
  }
  /**
   * Equivalent to `.min(1)`
   */
  nonempty(e) {
    return this.min(1, m.errToObj(e));
  }
  trim() {
    return new Q({
      ...this._def,
      checks: [...this._def.checks, { kind: "trim" }]
    });
  }
  toLowerCase() {
    return new Q({
      ...this._def,
      checks: [...this._def.checks, { kind: "toLowerCase" }]
    });
  }
  toUpperCase() {
    return new Q({
      ...this._def,
      checks: [...this._def.checks, { kind: "toUpperCase" }]
    });
  }
  get isDatetime() {
    return !!this._def.checks.find((e) => e.kind === "datetime");
  }
  get isDate() {
    return !!this._def.checks.find((e) => e.kind === "date");
  }
  get isTime() {
    return !!this._def.checks.find((e) => e.kind === "time");
  }
  get isDuration() {
    return !!this._def.checks.find((e) => e.kind === "duration");
  }
  get isEmail() {
    return !!this._def.checks.find((e) => e.kind === "email");
  }
  get isURL() {
    return !!this._def.checks.find((e) => e.kind === "url");
  }
  get isEmoji() {
    return !!this._def.checks.find((e) => e.kind === "emoji");
  }
  get isUUID() {
    return !!this._def.checks.find((e) => e.kind === "uuid");
  }
  get isNANOID() {
    return !!this._def.checks.find((e) => e.kind === "nanoid");
  }
  get isCUID() {
    return !!this._def.checks.find((e) => e.kind === "cuid");
  }
  get isCUID2() {
    return !!this._def.checks.find((e) => e.kind === "cuid2");
  }
  get isULID() {
    return !!this._def.checks.find((e) => e.kind === "ulid");
  }
  get isIP() {
    return !!this._def.checks.find((e) => e.kind === "ip");
  }
  get isCIDR() {
    return !!this._def.checks.find((e) => e.kind === "cidr");
  }
  get isBase64() {
    return !!this._def.checks.find((e) => e.kind === "base64");
  }
  get isBase64url() {
    return !!this._def.checks.find((e) => e.kind === "base64url");
  }
  get minLength() {
    let e = null;
    for (const s of this._def.checks)
      s.kind === "min" && (e === null || s.value > e) && (e = s.value);
    return e;
  }
  get maxLength() {
    let e = null;
    for (const s of this._def.checks)
      s.kind === "max" && (e === null || s.value < e) && (e = s.value);
    return e;
  }
}
Q.create = (t) => new Q({
  checks: [],
  typeName: v.ZodString,
  coerce: (t == null ? void 0 : t.coerce) ?? !1,
  ..._(t)
});
function fr(t, e) {
  const s = (t.toString().split(".")[1] || "").length, r = (e.toString().split(".")[1] || "").length, a = s > r ? s : r, n = Number.parseInt(t.toFixed(a).replace(".", "")), i = Number.parseInt(e.toFixed(a).replace(".", ""));
  return n % i / 10 ** a;
}
class me extends k {
  constructor() {
    super(...arguments), this.min = this.gte, this.max = this.lte, this.step = this.multipleOf;
  }
  _parse(e) {
    if (this._def.coerce && (e.data = Number(e.data)), this._getType(e) !== h.number) {
      const n = this._getOrReturnCtx(e);
      return u(n, {
        code: c.invalid_type,
        expected: h.number,
        received: n.parsedType
      }), y;
    }
    let r;
    const a = new V();
    for (const n of this._def.checks)
      n.kind === "int" ? b.isInteger(e.data) || (r = this._getOrReturnCtx(e, r), u(r, {
        code: c.invalid_type,
        expected: "integer",
        received: "float",
        message: n.message
      }), a.dirty()) : n.kind === "min" ? (n.inclusive ? e.data < n.value : e.data <= n.value) && (r = this._getOrReturnCtx(e, r), u(r, {
        code: c.too_small,
        minimum: n.value,
        type: "number",
        inclusive: n.inclusive,
        exact: !1,
        message: n.message
      }), a.dirty()) : n.kind === "max" ? (n.inclusive ? e.data > n.value : e.data >= n.value) && (r = this._getOrReturnCtx(e, r), u(r, {
        code: c.too_big,
        maximum: n.value,
        type: "number",
        inclusive: n.inclusive,
        exact: !1,
        message: n.message
      }), a.dirty()) : n.kind === "multipleOf" ? fr(e.data, n.value) !== 0 && (r = this._getOrReturnCtx(e, r), u(r, {
        code: c.not_multiple_of,
        multipleOf: n.value,
        message: n.message
      }), a.dirty()) : n.kind === "finite" ? Number.isFinite(e.data) || (r = this._getOrReturnCtx(e, r), u(r, {
        code: c.not_finite,
        message: n.message
      }), a.dirty()) : b.assertNever(n);
    return { status: a.value, value: e.data };
  }
  gte(e, s) {
    return this.setLimit("min", e, !0, m.toString(s));
  }
  gt(e, s) {
    return this.setLimit("min", e, !1, m.toString(s));
  }
  lte(e, s) {
    return this.setLimit("max", e, !0, m.toString(s));
  }
  lt(e, s) {
    return this.setLimit("max", e, !1, m.toString(s));
  }
  setLimit(e, s, r, a) {
    return new me({
      ...this._def,
      checks: [
        ...this._def.checks,
        {
          kind: e,
          value: s,
          inclusive: r,
          message: m.toString(a)
        }
      ]
    });
  }
  _addCheck(e) {
    return new me({
      ...this._def,
      checks: [...this._def.checks, e]
    });
  }
  int(e) {
    return this._addCheck({
      kind: "int",
      message: m.toString(e)
    });
  }
  positive(e) {
    return this._addCheck({
      kind: "min",
      value: 0,
      inclusive: !1,
      message: m.toString(e)
    });
  }
  negative(e) {
    return this._addCheck({
      kind: "max",
      value: 0,
      inclusive: !1,
      message: m.toString(e)
    });
  }
  nonpositive(e) {
    return this._addCheck({
      kind: "max",
      value: 0,
      inclusive: !0,
      message: m.toString(e)
    });
  }
  nonnegative(e) {
    return this._addCheck({
      kind: "min",
      value: 0,
      inclusive: !0,
      message: m.toString(e)
    });
  }
  multipleOf(e, s) {
    return this._addCheck({
      kind: "multipleOf",
      value: e,
      message: m.toString(s)
    });
  }
  finite(e) {
    return this._addCheck({
      kind: "finite",
      message: m.toString(e)
    });
  }
  safe(e) {
    return this._addCheck({
      kind: "min",
      inclusive: !0,
      value: Number.MIN_SAFE_INTEGER,
      message: m.toString(e)
    })._addCheck({
      kind: "max",
      inclusive: !0,
      value: Number.MAX_SAFE_INTEGER,
      message: m.toString(e)
    });
  }
  get minValue() {
    let e = null;
    for (const s of this._def.checks)
      s.kind === "min" && (e === null || s.value > e) && (e = s.value);
    return e;
  }
  get maxValue() {
    let e = null;
    for (const s of this._def.checks)
      s.kind === "max" && (e === null || s.value < e) && (e = s.value);
    return e;
  }
  get isInt() {
    return !!this._def.checks.find((e) => e.kind === "int" || e.kind === "multipleOf" && b.isInteger(e.value));
  }
  get isFinite() {
    let e = null, s = null;
    for (const r of this._def.checks) {
      if (r.kind === "finite" || r.kind === "int" || r.kind === "multipleOf")
        return !0;
      r.kind === "min" ? (s === null || r.value > s) && (s = r.value) : r.kind === "max" && (e === null || r.value < e) && (e = r.value);
    }
    return Number.isFinite(s) && Number.isFinite(e);
  }
}
me.create = (t) => new me({
  checks: [],
  typeName: v.ZodNumber,
  coerce: (t == null ? void 0 : t.coerce) || !1,
  ..._(t)
});
class pe extends k {
  constructor() {
    super(...arguments), this.min = this.gte, this.max = this.lte;
  }
  _parse(e) {
    if (this._def.coerce)
      try {
        e.data = BigInt(e.data);
      } catch {
        return this._getInvalidInput(e);
      }
    if (this._getType(e) !== h.bigint)
      return this._getInvalidInput(e);
    let r;
    const a = new V();
    for (const n of this._def.checks)
      n.kind === "min" ? (n.inclusive ? e.data < n.value : e.data <= n.value) && (r = this._getOrReturnCtx(e, r), u(r, {
        code: c.too_small,
        type: "bigint",
        minimum: n.value,
        inclusive: n.inclusive,
        message: n.message
      }), a.dirty()) : n.kind === "max" ? (n.inclusive ? e.data > n.value : e.data >= n.value) && (r = this._getOrReturnCtx(e, r), u(r, {
        code: c.too_big,
        type: "bigint",
        maximum: n.value,
        inclusive: n.inclusive,
        message: n.message
      }), a.dirty()) : n.kind === "multipleOf" ? e.data % n.value !== BigInt(0) && (r = this._getOrReturnCtx(e, r), u(r, {
        code: c.not_multiple_of,
        multipleOf: n.value,
        message: n.message
      }), a.dirty()) : b.assertNever(n);
    return { status: a.value, value: e.data };
  }
  _getInvalidInput(e) {
    const s = this._getOrReturnCtx(e);
    return u(s, {
      code: c.invalid_type,
      expected: h.bigint,
      received: s.parsedType
    }), y;
  }
  gte(e, s) {
    return this.setLimit("min", e, !0, m.toString(s));
  }
  gt(e, s) {
    return this.setLimit("min", e, !1, m.toString(s));
  }
  lte(e, s) {
    return this.setLimit("max", e, !0, m.toString(s));
  }
  lt(e, s) {
    return this.setLimit("max", e, !1, m.toString(s));
  }
  setLimit(e, s, r, a) {
    return new pe({
      ...this._def,
      checks: [
        ...this._def.checks,
        {
          kind: e,
          value: s,
          inclusive: r,
          message: m.toString(a)
        }
      ]
    });
  }
  _addCheck(e) {
    return new pe({
      ...this._def,
      checks: [...this._def.checks, e]
    });
  }
  positive(e) {
    return this._addCheck({
      kind: "min",
      value: BigInt(0),
      inclusive: !1,
      message: m.toString(e)
    });
  }
  negative(e) {
    return this._addCheck({
      kind: "max",
      value: BigInt(0),
      inclusive: !1,
      message: m.toString(e)
    });
  }
  nonpositive(e) {
    return this._addCheck({
      kind: "max",
      value: BigInt(0),
      inclusive: !0,
      message: m.toString(e)
    });
  }
  nonnegative(e) {
    return this._addCheck({
      kind: "min",
      value: BigInt(0),
      inclusive: !0,
      message: m.toString(e)
    });
  }
  multipleOf(e, s) {
    return this._addCheck({
      kind: "multipleOf",
      value: e,
      message: m.toString(s)
    });
  }
  get minValue() {
    let e = null;
    for (const s of this._def.checks)
      s.kind === "min" && (e === null || s.value > e) && (e = s.value);
    return e;
  }
  get maxValue() {
    let e = null;
    for (const s of this._def.checks)
      s.kind === "max" && (e === null || s.value < e) && (e = s.value);
    return e;
  }
}
pe.create = (t) => new pe({
  checks: [],
  typeName: v.ZodBigInt,
  coerce: (t == null ? void 0 : t.coerce) ?? !1,
  ..._(t)
});
class lt extends k {
  _parse(e) {
    if (this._def.coerce && (e.data = !!e.data), this._getType(e) !== h.boolean) {
      const r = this._getOrReturnCtx(e);
      return u(r, {
        code: c.invalid_type,
        expected: h.boolean,
        received: r.parsedType
      }), y;
    }
    return B(e.data);
  }
}
lt.create = (t) => new lt({
  typeName: v.ZodBoolean,
  coerce: (t == null ? void 0 : t.coerce) || !1,
  ..._(t)
});
class we extends k {
  _parse(e) {
    if (this._def.coerce && (e.data = new Date(e.data)), this._getType(e) !== h.date) {
      const n = this._getOrReturnCtx(e);
      return u(n, {
        code: c.invalid_type,
        expected: h.date,
        received: n.parsedType
      }), y;
    }
    if (Number.isNaN(e.data.getTime())) {
      const n = this._getOrReturnCtx(e);
      return u(n, {
        code: c.invalid_date
      }), y;
    }
    const r = new V();
    let a;
    for (const n of this._def.checks)
      n.kind === "min" ? e.data.getTime() < n.value && (a = this._getOrReturnCtx(e, a), u(a, {
        code: c.too_small,
        message: n.message,
        inclusive: !0,
        exact: !1,
        minimum: n.value,
        type: "date"
      }), r.dirty()) : n.kind === "max" ? e.data.getTime() > n.value && (a = this._getOrReturnCtx(e, a), u(a, {
        code: c.too_big,
        message: n.message,
        inclusive: !0,
        exact: !1,
        maximum: n.value,
        type: "date"
      }), r.dirty()) : b.assertNever(n);
    return {
      status: r.value,
      value: new Date(e.data.getTime())
    };
  }
  _addCheck(e) {
    return new we({
      ...this._def,
      checks: [...this._def.checks, e]
    });
  }
  min(e, s) {
    return this._addCheck({
      kind: "min",
      value: e.getTime(),
      message: m.toString(s)
    });
  }
  max(e, s) {
    return this._addCheck({
      kind: "max",
      value: e.getTime(),
      message: m.toString(s)
    });
  }
  get minDate() {
    let e = null;
    for (const s of this._def.checks)
      s.kind === "min" && (e === null || s.value > e) && (e = s.value);
    return e != null ? new Date(e) : null;
  }
  get maxDate() {
    let e = null;
    for (const s of this._def.checks)
      s.kind === "max" && (e === null || s.value < e) && (e = s.value);
    return e != null ? new Date(e) : null;
  }
}
we.create = (t) => new we({
  checks: [],
  coerce: (t == null ? void 0 : t.coerce) || !1,
  typeName: v.ZodDate,
  ..._(t)
});
class dt extends k {
  _parse(e) {
    if (this._getType(e) !== h.symbol) {
      const r = this._getOrReturnCtx(e);
      return u(r, {
        code: c.invalid_type,
        expected: h.symbol,
        received: r.parsedType
      }), y;
    }
    return B(e.data);
  }
}
dt.create = (t) => new dt({
  typeName: v.ZodSymbol,
  ..._(t)
});
class ct extends k {
  _parse(e) {
    if (this._getType(e) !== h.undefined) {
      const r = this._getOrReturnCtx(e);
      return u(r, {
        code: c.invalid_type,
        expected: h.undefined,
        received: r.parsedType
      }), y;
    }
    return B(e.data);
  }
}
ct.create = (t) => new ct({
  typeName: v.ZodUndefined,
  ..._(t)
});
class ut extends k {
  _parse(e) {
    if (this._getType(e) !== h.null) {
      const r = this._getOrReturnCtx(e);
      return u(r, {
        code: c.invalid_type,
        expected: h.null,
        received: r.parsedType
      }), y;
    }
    return B(e.data);
  }
}
ut.create = (t) => new ut({
  typeName: v.ZodNull,
  ..._(t)
});
class ft extends k {
  constructor() {
    super(...arguments), this._any = !0;
  }
  _parse(e) {
    return B(e.data);
  }
}
ft.create = (t) => new ft({
  typeName: v.ZodAny,
  ..._(t)
});
class ht extends k {
  constructor() {
    super(...arguments), this._unknown = !0;
  }
  _parse(e) {
    return B(e.data);
  }
}
ht.create = (t) => new ht({
  typeName: v.ZodUnknown,
  ..._(t)
});
class ee extends k {
  _parse(e) {
    const s = this._getOrReturnCtx(e);
    return u(s, {
      code: c.invalid_type,
      expected: h.never,
      received: s.parsedType
    }), y;
  }
}
ee.create = (t) => new ee({
  typeName: v.ZodNever,
  ..._(t)
});
class mt extends k {
  _parse(e) {
    if (this._getType(e) !== h.undefined) {
      const r = this._getOrReturnCtx(e);
      return u(r, {
        code: c.invalid_type,
        expected: h.void,
        received: r.parsedType
      }), y;
    }
    return B(e.data);
  }
}
mt.create = (t) => new mt({
  typeName: v.ZodVoid,
  ..._(t)
});
class U extends k {
  _parse(e) {
    const { ctx: s, status: r } = this._processInputParams(e), a = this._def;
    if (s.parsedType !== h.array)
      return u(s, {
        code: c.invalid_type,
        expected: h.array,
        received: s.parsedType
      }), y;
    if (a.exactLength !== null) {
      const i = s.data.length > a.exactLength.value, o = s.data.length < a.exactLength.value;
      (i || o) && (u(s, {
        code: i ? c.too_big : c.too_small,
        minimum: o ? a.exactLength.value : void 0,
        maximum: i ? a.exactLength.value : void 0,
        type: "array",
        inclusive: !0,
        exact: !0,
        message: a.exactLength.message
      }), r.dirty());
    }
    if (a.minLength !== null && s.data.length < a.minLength.value && (u(s, {
      code: c.too_small,
      minimum: a.minLength.value,
      type: "array",
      inclusive: !0,
      exact: !1,
      message: a.minLength.message
    }), r.dirty()), a.maxLength !== null && s.data.length > a.maxLength.value && (u(s, {
      code: c.too_big,
      maximum: a.maxLength.value,
      type: "array",
      inclusive: !0,
      exact: !1,
      message: a.maxLength.message
    }), r.dirty()), s.common.async)
      return Promise.all([...s.data].map((i, o) => a.type._parseAsync(new K(s, i, s.path, o)))).then((i) => V.mergeArray(r, i));
    const n = [...s.data].map((i, o) => a.type._parseSync(new K(s, i, s.path, o)));
    return V.mergeArray(r, n);
  }
  get element() {
    return this._def.type;
  }
  min(e, s) {
    return new U({
      ...this._def,
      minLength: { value: e, message: m.toString(s) }
    });
  }
  max(e, s) {
    return new U({
      ...this._def,
      maxLength: { value: e, message: m.toString(s) }
    });
  }
  length(e, s) {
    return new U({
      ...this._def,
      exactLength: { value: e, message: m.toString(s) }
    });
  }
  nonempty(e) {
    return this.min(1, e);
  }
}
U.create = (t, e) => new U({
  type: t,
  minLength: null,
  maxLength: null,
  exactLength: null,
  typeName: v.ZodArray,
  ..._(e)
});
function ie(t) {
  if (t instanceof S) {
    const e = {};
    for (const s in t.shape) {
      const r = t.shape[s];
      e[s] = G.create(ie(r));
    }
    return new S({
      ...t._def,
      shape: () => e
    });
  } else return t instanceof U ? new U({
    ...t._def,
    type: ie(t.element)
  }) : t instanceof G ? G.create(ie(t.unwrap())) : t instanceof ue ? ue.create(ie(t.unwrap())) : t instanceof ae ? ae.create(t.items.map((e) => ie(e))) : t;
}
class S extends k {
  constructor() {
    super(...arguments), this._cached = null, this.nonstrict = this.passthrough, this.augment = this.extend;
  }
  _getCached() {
    if (this._cached !== null)
      return this._cached;
    const e = this._def.shape(), s = b.objectKeys(e);
    return this._cached = { shape: e, keys: s }, this._cached;
  }
  _parse(e) {
    if (this._getType(e) !== h.object) {
      const d = this._getOrReturnCtx(e);
      return u(d, {
        code: c.invalid_type,
        expected: h.object,
        received: d.parsedType
      }), y;
    }
    const { status: r, ctx: a } = this._processInputParams(e), { shape: n, keys: i } = this._getCached(), o = [];
    if (!(this._def.catchall instanceof ee && this._def.unknownKeys === "strip"))
      for (const d in a.data)
        i.includes(d) || o.push(d);
    const l = [];
    for (const d of i) {
      const f = n[d], E = a.data[d];
      l.push({
        key: { status: "valid", value: d },
        value: f._parse(new K(a, E, a.path, d)),
        alwaysSet: d in a.data
      });
    }
    if (this._def.catchall instanceof ee) {
      const d = this._def.unknownKeys;
      if (d === "passthrough")
        for (const f of o)
          l.push({
            key: { status: "valid", value: f },
            value: { status: "valid", value: a.data[f] }
          });
      else if (d === "strict")
        o.length > 0 && (u(a, {
          code: c.unrecognized_keys,
          keys: o
        }), r.dirty());
      else if (d !== "strip") throw new Error("Internal ZodObject error: invalid unknownKeys value.");
    } else {
      const d = this._def.catchall;
      for (const f of o) {
        const E = a.data[f];
        l.push({
          key: { status: "valid", value: f },
          value: d._parse(
            new K(a, E, a.path, f)
            //, ctx.child(key), value, getParsedType(value)
          ),
          alwaysSet: f in a.data
        });
      }
    }
    return a.common.async ? Promise.resolve().then(async () => {
      const d = [];
      for (const f of l) {
        const E = await f.key, Z = await f.value;
        d.push({
          key: E,
          value: Z,
          alwaysSet: f.alwaysSet
        });
      }
      return d;
    }).then((d) => V.mergeObjectSync(r, d)) : V.mergeObjectSync(r, l);
  }
  get shape() {
    return this._def.shape();
  }
  strict(e) {
    return m.errToObj, new S({
      ...this._def,
      unknownKeys: "strict",
      ...e !== void 0 ? {
        errorMap: (s, r) => {
          var n, i;
          const a = ((i = (n = this._def).errorMap) == null ? void 0 : i.call(n, s, r).message) ?? r.defaultError;
          return s.code === "unrecognized_keys" ? {
            message: m.errToObj(e).message ?? a
          } : {
            message: a
          };
        }
      } : {}
    });
  }
  strip() {
    return new S({
      ...this._def,
      unknownKeys: "strip"
    });
  }
  passthrough() {
    return new S({
      ...this._def,
      unknownKeys: "passthrough"
    });
  }
  // const AugmentFactory =
  //   <Def extends ZodObjectDef>(def: Def) =>
  //   <Augmentation extends ZodRawShape>(
  //     augmentation: Augmentation
  //   ): ZodObject<
  //     extendShape<ReturnType<Def["shape"]>, Augmentation>,
  //     Def["unknownKeys"],
  //     Def["catchall"]
  //   > => {
  //     return new ZodObject({
  //       ...def,
  //       shape: () => ({
  //         ...def.shape(),
  //         ...augmentation,
  //       }),
  //     }) as any;
  //   };
  extend(e) {
    return new S({
      ...this._def,
      shape: () => ({
        ...this._def.shape(),
        ...e
      })
    });
  }
  /**
   * Prior to zod@1.0.12 there was a bug in the
   * inferred type of merged objects. Please
   * upgrade if you are experiencing issues.
   */
  merge(e) {
    return new S({
      unknownKeys: e._def.unknownKeys,
      catchall: e._def.catchall,
      shape: () => ({
        ...this._def.shape(),
        ...e._def.shape()
      }),
      typeName: v.ZodObject
    });
  }
  // merge<
  //   Incoming extends AnyZodObject,
  //   Augmentation extends Incoming["shape"],
  //   NewOutput extends {
  //     [k in keyof Augmentation | keyof Output]: k extends keyof Augmentation
  //       ? Augmentation[k]["_output"]
  //       : k extends keyof Output
  //       ? Output[k]
  //       : never;
  //   },
  //   NewInput extends {
  //     [k in keyof Augmentation | keyof Input]: k extends keyof Augmentation
  //       ? Augmentation[k]["_input"]
  //       : k extends keyof Input
  //       ? Input[k]
  //       : never;
  //   }
  // >(
  //   merging: Incoming
  // ): ZodObject<
  //   extendShape<T, ReturnType<Incoming["_def"]["shape"]>>,
  //   Incoming["_def"]["unknownKeys"],
  //   Incoming["_def"]["catchall"],
  //   NewOutput,
  //   NewInput
  // > {
  //   const merged: any = new ZodObject({
  //     unknownKeys: merging._def.unknownKeys,
  //     catchall: merging._def.catchall,
  //     shape: () =>
  //       objectUtil.mergeShapes(this._def.shape(), merging._def.shape()),
  //     typeName: ZodFirstPartyTypeKind.ZodObject,
  //   }) as any;
  //   return merged;
  // }
  setKey(e, s) {
    return this.augment({ [e]: s });
  }
  // merge<Incoming extends AnyZodObject>(
  //   merging: Incoming
  // ): //ZodObject<T & Incoming["_shape"], UnknownKeys, Catchall> = (merging) => {
  // ZodObject<
  //   extendShape<T, ReturnType<Incoming["_def"]["shape"]>>,
  //   Incoming["_def"]["unknownKeys"],
  //   Incoming["_def"]["catchall"]
  // > {
  //   // const mergedShape = objectUtil.mergeShapes(
  //   //   this._def.shape(),
  //   //   merging._def.shape()
  //   // );
  //   const merged: any = new ZodObject({
  //     unknownKeys: merging._def.unknownKeys,
  //     catchall: merging._def.catchall,
  //     shape: () =>
  //       objectUtil.mergeShapes(this._def.shape(), merging._def.shape()),
  //     typeName: ZodFirstPartyTypeKind.ZodObject,
  //   }) as any;
  //   return merged;
  // }
  catchall(e) {
    return new S({
      ...this._def,
      catchall: e
    });
  }
  pick(e) {
    const s = {};
    for (const r of b.objectKeys(e))
      e[r] && this.shape[r] && (s[r] = this.shape[r]);
    return new S({
      ...this._def,
      shape: () => s
    });
  }
  omit(e) {
    const s = {};
    for (const r of b.objectKeys(this.shape))
      e[r] || (s[r] = this.shape[r]);
    return new S({
      ...this._def,
      shape: () => s
    });
  }
  /**
   * @deprecated
   */
  deepPartial() {
    return ie(this);
  }
  partial(e) {
    const s = {};
    for (const r of b.objectKeys(this.shape)) {
      const a = this.shape[r];
      e && !e[r] ? s[r] = a : s[r] = a.optional();
    }
    return new S({
      ...this._def,
      shape: () => s
    });
  }
  required(e) {
    const s = {};
    for (const r of b.objectKeys(this.shape))
      if (e && !e[r])
        s[r] = this.shape[r];
      else {
        let n = this.shape[r];
        for (; n instanceof G; )
          n = n._def.innerType;
        s[r] = n;
      }
    return new S({
      ...this._def,
      shape: () => s
    });
  }
  keyof() {
    return Kt(b.objectKeys(this.shape));
  }
}
S.create = (t, e) => new S({
  shape: () => t,
  unknownKeys: "strip",
  catchall: ee.create(),
  typeName: v.ZodObject,
  ..._(e)
});
S.strictCreate = (t, e) => new S({
  shape: () => t,
  unknownKeys: "strict",
  catchall: ee.create(),
  typeName: v.ZodObject,
  ..._(e)
});
S.lazycreate = (t, e) => new S({
  shape: t,
  unknownKeys: "strip",
  catchall: ee.create(),
  typeName: v.ZodObject,
  ..._(e)
});
class Te extends k {
  _parse(e) {
    const { ctx: s } = this._processInputParams(e), r = this._def.options;
    function a(n) {
      for (const o of n)
        if (o.result.status === "valid")
          return o.result;
      for (const o of n)
        if (o.result.status === "dirty")
          return s.common.issues.push(...o.ctx.common.issues), o.result;
      const i = n.map((o) => new q(o.ctx.common.issues));
      return u(s, {
        code: c.invalid_union,
        unionErrors: i
      }), y;
    }
    if (s.common.async)
      return Promise.all(r.map(async (n) => {
        const i = {
          ...s,
          common: {
            ...s.common,
            issues: []
          },
          parent: null
        };
        return {
          result: await n._parseAsync({
            data: s.data,
            path: s.path,
            parent: i
          }),
          ctx: i
        };
      })).then(a);
    {
      let n;
      const i = [];
      for (const l of r) {
        const d = {
          ...s,
          common: {
            ...s.common,
            issues: []
          },
          parent: null
        }, f = l._parseSync({
          data: s.data,
          path: s.path,
          parent: d
        });
        if (f.status === "valid")
          return f;
        f.status === "dirty" && !n && (n = { result: f, ctx: d }), d.common.issues.length && i.push(d.common.issues);
      }
      if (n)
        return s.common.issues.push(...n.ctx.common.issues), n.result;
      const o = i.map((l) => new q(l));
      return u(s, {
        code: c.invalid_union,
        unionErrors: o
      }), y;
    }
  }
  get options() {
    return this._def.options;
  }
}
Te.create = (t, e) => new Te({
  options: t,
  typeName: v.ZodUnion,
  ..._(e)
});
function De(t, e) {
  const s = J(t), r = J(e);
  if (t === e)
    return { valid: !0, data: t };
  if (s === h.object && r === h.object) {
    const a = b.objectKeys(e), n = b.objectKeys(t).filter((o) => a.indexOf(o) !== -1), i = { ...t, ...e };
    for (const o of n) {
      const l = De(t[o], e[o]);
      if (!l.valid)
        return { valid: !1 };
      i[o] = l.data;
    }
    return { valid: !0, data: i };
  } else if (s === h.array && r === h.array) {
    if (t.length !== e.length)
      return { valid: !1 };
    const a = [];
    for (let n = 0; n < t.length; n++) {
      const i = t[n], o = e[n], l = De(i, o);
      if (!l.valid)
        return { valid: !1 };
      a.push(l.data);
    }
    return { valid: !0, data: a };
  } else return s === h.date && r === h.date && +t == +e ? { valid: !0, data: t } : { valid: !1 };
}
class Ce extends k {
  _parse(e) {
    const { status: s, ctx: r } = this._processInputParams(e), a = (n, i) => {
      if (nt(n) || nt(i))
        return y;
      const o = De(n.value, i.value);
      return o.valid ? ((it(n) || it(i)) && s.dirty(), { status: s.value, value: o.data }) : (u(r, {
        code: c.invalid_intersection_types
      }), y);
    };
    return r.common.async ? Promise.all([
      this._def.left._parseAsync({
        data: r.data,
        path: r.path,
        parent: r
      }),
      this._def.right._parseAsync({
        data: r.data,
        path: r.path,
        parent: r
      })
    ]).then(([n, i]) => a(n, i)) : a(this._def.left._parseSync({
      data: r.data,
      path: r.path,
      parent: r
    }), this._def.right._parseSync({
      data: r.data,
      path: r.path,
      parent: r
    }));
  }
}
Ce.create = (t, e, s) => new Ce({
  left: t,
  right: e,
  typeName: v.ZodIntersection,
  ..._(s)
});
class ae extends k {
  _parse(e) {
    const { status: s, ctx: r } = this._processInputParams(e);
    if (r.parsedType !== h.array)
      return u(r, {
        code: c.invalid_type,
        expected: h.array,
        received: r.parsedType
      }), y;
    if (r.data.length < this._def.items.length)
      return u(r, {
        code: c.too_small,
        minimum: this._def.items.length,
        inclusive: !0,
        exact: !1,
        type: "array"
      }), y;
    !this._def.rest && r.data.length > this._def.items.length && (u(r, {
      code: c.too_big,
      maximum: this._def.items.length,
      inclusive: !0,
      exact: !1,
      type: "array"
    }), s.dirty());
    const n = [...r.data].map((i, o) => {
      const l = this._def.items[o] || this._def.rest;
      return l ? l._parse(new K(r, i, r.path, o)) : null;
    }).filter((i) => !!i);
    return r.common.async ? Promise.all(n).then((i) => V.mergeArray(s, i)) : V.mergeArray(s, n);
  }
  get items() {
    return this._def.items;
  }
  rest(e) {
    return new ae({
      ...this._def,
      rest: e
    });
  }
}
ae.create = (t, e) => {
  if (!Array.isArray(t))
    throw new Error("You must pass an array of schemas to z.tuple([ ... ])");
  return new ae({
    items: t,
    typeName: v.ZodTuple,
    rest: null,
    ..._(e)
  });
};
class pt extends k {
  get keySchema() {
    return this._def.keyType;
  }
  get valueSchema() {
    return this._def.valueType;
  }
  _parse(e) {
    const { status: s, ctx: r } = this._processInputParams(e);
    if (r.parsedType !== h.map)
      return u(r, {
        code: c.invalid_type,
        expected: h.map,
        received: r.parsedType
      }), y;
    const a = this._def.keyType, n = this._def.valueType, i = [...r.data.entries()].map(([o, l], d) => ({
      key: a._parse(new K(r, o, r.path, [d, "key"])),
      value: n._parse(new K(r, l, r.path, [d, "value"]))
    }));
    if (r.common.async) {
      const o = /* @__PURE__ */ new Map();
      return Promise.resolve().then(async () => {
        for (const l of i) {
          const d = await l.key, f = await l.value;
          if (d.status === "aborted" || f.status === "aborted")
            return y;
          (d.status === "dirty" || f.status === "dirty") && s.dirty(), o.set(d.value, f.value);
        }
        return { status: s.value, value: o };
      });
    } else {
      const o = /* @__PURE__ */ new Map();
      for (const l of i) {
        const d = l.key, f = l.value;
        if (d.status === "aborted" || f.status === "aborted")
          return y;
        (d.status === "dirty" || f.status === "dirty") && s.dirty(), o.set(d.value, f.value);
      }
      return { status: s.value, value: o };
    }
  }
}
pt.create = (t, e, s) => new pt({
  valueType: e,
  keyType: t,
  typeName: v.ZodMap,
  ..._(s)
});
class ge extends k {
  _parse(e) {
    const { status: s, ctx: r } = this._processInputParams(e);
    if (r.parsedType !== h.set)
      return u(r, {
        code: c.invalid_type,
        expected: h.set,
        received: r.parsedType
      }), y;
    const a = this._def;
    a.minSize !== null && r.data.size < a.minSize.value && (u(r, {
      code: c.too_small,
      minimum: a.minSize.value,
      type: "set",
      inclusive: !0,
      exact: !1,
      message: a.minSize.message
    }), s.dirty()), a.maxSize !== null && r.data.size > a.maxSize.value && (u(r, {
      code: c.too_big,
      maximum: a.maxSize.value,
      type: "set",
      inclusive: !0,
      exact: !1,
      message: a.maxSize.message
    }), s.dirty());
    const n = this._def.valueType;
    function i(l) {
      const d = /* @__PURE__ */ new Set();
      for (const f of l) {
        if (f.status === "aborted")
          return y;
        f.status === "dirty" && s.dirty(), d.add(f.value);
      }
      return { status: s.value, value: d };
    }
    const o = [...r.data.values()].map((l, d) => n._parse(new K(r, l, r.path, d)));
    return r.common.async ? Promise.all(o).then((l) => i(l)) : i(o);
  }
  min(e, s) {
    return new ge({
      ...this._def,
      minSize: { value: e, message: m.toString(s) }
    });
  }
  max(e, s) {
    return new ge({
      ...this._def,
      maxSize: { value: e, message: m.toString(s) }
    });
  }
  size(e, s) {
    return this.min(e, s).max(e, s);
  }
  nonempty(e) {
    return this.min(1, e);
  }
}
ge.create = (t, e) => new ge({
  valueType: t,
  minSize: null,
  maxSize: null,
  typeName: v.ZodSet,
  ..._(e)
});
class gt extends k {
  get schema() {
    return this._def.getter();
  }
  _parse(e) {
    const { ctx: s } = this._processInputParams(e);
    return this._def.getter()._parse({ data: s.data, path: s.path, parent: s });
  }
}
gt.create = (t, e) => new gt({
  getter: t,
  typeName: v.ZodLazy,
  ..._(e)
});
class yt extends k {
  _parse(e) {
    if (e.data !== this._def.value) {
      const s = this._getOrReturnCtx(e);
      return u(s, {
        received: s.data,
        code: c.invalid_literal,
        expected: this._def.value
      }), y;
    }
    return { status: "valid", value: e.data };
  }
  get value() {
    return this._def.value;
  }
}
yt.create = (t, e) => new yt({
  value: t,
  typeName: v.ZodLiteral,
  ..._(e)
});
function Kt(t, e) {
  return new de({
    values: t,
    typeName: v.ZodEnum,
    ..._(e)
  });
}
class de extends k {
  _parse(e) {
    if (typeof e.data != "string") {
      const s = this._getOrReturnCtx(e), r = this._def.values;
      return u(s, {
        expected: b.joinValues(r),
        received: s.parsedType,
        code: c.invalid_type
      }), y;
    }
    if (this._cache || (this._cache = new Set(this._def.values)), !this._cache.has(e.data)) {
      const s = this._getOrReturnCtx(e), r = this._def.values;
      return u(s, {
        received: s.data,
        code: c.invalid_enum_value,
        options: r
      }), y;
    }
    return B(e.data);
  }
  get options() {
    return this._def.values;
  }
  get enum() {
    const e = {};
    for (const s of this._def.values)
      e[s] = s;
    return e;
  }
  get Values() {
    const e = {};
    for (const s of this._def.values)
      e[s] = s;
    return e;
  }
  get Enum() {
    const e = {};
    for (const s of this._def.values)
      e[s] = s;
    return e;
  }
  extract(e, s = this._def) {
    return de.create(e, {
      ...this._def,
      ...s
    });
  }
  exclude(e, s = this._def) {
    return de.create(this.options.filter((r) => !e.includes(r)), {
      ...this._def,
      ...s
    });
  }
}
de.create = Kt;
class vt extends k {
  _parse(e) {
    const s = b.getValidEnumValues(this._def.values), r = this._getOrReturnCtx(e);
    if (r.parsedType !== h.string && r.parsedType !== h.number) {
      const a = b.objectValues(s);
      return u(r, {
        expected: b.joinValues(a),
        received: r.parsedType,
        code: c.invalid_type
      }), y;
    }
    if (this._cache || (this._cache = new Set(b.getValidEnumValues(this._def.values))), !this._cache.has(e.data)) {
      const a = b.objectValues(s);
      return u(r, {
        received: r.data,
        code: c.invalid_enum_value,
        options: a
      }), y;
    }
    return B(e.data);
  }
  get enum() {
    return this._def.values;
  }
}
vt.create = (t, e) => new vt({
  values: t,
  typeName: v.ZodNativeEnum,
  ..._(e)
});
class Ne extends k {
  unwrap() {
    return this._def.type;
  }
  _parse(e) {
    const { ctx: s } = this._processInputParams(e);
    if (s.parsedType !== h.promise && s.common.async === !1)
      return u(s, {
        code: c.invalid_type,
        expected: h.promise,
        received: s.parsedType
      }), y;
    const r = s.parsedType === h.promise ? s.data : Promise.resolve(s.data);
    return B(r.then((a) => this._def.type.parseAsync(a, {
      path: s.path,
      errorMap: s.common.contextualErrorMap
    })));
  }
}
Ne.create = (t, e) => new Ne({
  type: t,
  typeName: v.ZodPromise,
  ..._(e)
});
class ce extends k {
  innerType() {
    return this._def.schema;
  }
  sourceType() {
    return this._def.schema._def.typeName === v.ZodEffects ? this._def.schema.sourceType() : this._def.schema;
  }
  _parse(e) {
    const { status: s, ctx: r } = this._processInputParams(e), a = this._def.effect || null, n = {
      addIssue: (i) => {
        u(r, i), i.fatal ? s.abort() : s.dirty();
      },
      get path() {
        return r.path;
      }
    };
    if (n.addIssue = n.addIssue.bind(n), a.type === "preprocess") {
      const i = a.transform(r.data, n);
      if (r.common.async)
        return Promise.resolve(i).then(async (o) => {
          if (s.value === "aborted")
            return y;
          const l = await this._def.schema._parseAsync({
            data: o,
            path: r.path,
            parent: r
          });
          return l.status === "aborted" ? y : l.status === "dirty" || s.value === "dirty" ? fe(l.value) : l;
        });
      {
        if (s.value === "aborted")
          return y;
        const o = this._def.schema._parseSync({
          data: i,
          path: r.path,
          parent: r
        });
        return o.status === "aborted" ? y : o.status === "dirty" || s.value === "dirty" ? fe(o.value) : o;
      }
    }
    if (a.type === "refinement") {
      const i = (o) => {
        const l = a.refinement(o, n);
        if (r.common.async)
          return Promise.resolve(l);
        if (l instanceof Promise)
          throw new Error("Async refinement encountered during synchronous parse operation. Use .parseAsync instead.");
        return o;
      };
      if (r.common.async === !1) {
        const o = this._def.schema._parseSync({
          data: r.data,
          path: r.path,
          parent: r
        });
        return o.status === "aborted" ? y : (o.status === "dirty" && s.dirty(), i(o.value), { status: s.value, value: o.value });
      } else
        return this._def.schema._parseAsync({ data: r.data, path: r.path, parent: r }).then((o) => o.status === "aborted" ? y : (o.status === "dirty" && s.dirty(), i(o.value).then(() => ({ status: s.value, value: o.value }))));
    }
    if (a.type === "transform")
      if (r.common.async === !1) {
        const i = this._def.schema._parseSync({
          data: r.data,
          path: r.path,
          parent: r
        });
        if (!le(i))
          return y;
        const o = a.transform(i.value, n);
        if (o instanceof Promise)
          throw new Error("Asynchronous transform encountered during synchronous parse operation. Use .parseAsync instead.");
        return { status: s.value, value: o };
      } else
        return this._def.schema._parseAsync({ data: r.data, path: r.path, parent: r }).then((i) => le(i) ? Promise.resolve(a.transform(i.value, n)).then((o) => ({
          status: s.value,
          value: o
        })) : y);
    b.assertNever(a);
  }
}
ce.create = (t, e, s) => new ce({
  schema: t,
  typeName: v.ZodEffects,
  effect: e,
  ..._(s)
});
ce.createWithPreprocess = (t, e, s) => new ce({
  schema: e,
  effect: { type: "preprocess", transform: t },
  typeName: v.ZodEffects,
  ..._(s)
});
class G extends k {
  _parse(e) {
    return this._getType(e) === h.undefined ? B(void 0) : this._def.innerType._parse(e);
  }
  unwrap() {
    return this._def.innerType;
  }
}
G.create = (t, e) => new G({
  innerType: t,
  typeName: v.ZodOptional,
  ..._(e)
});
class ue extends k {
  _parse(e) {
    return this._getType(e) === h.null ? B(null) : this._def.innerType._parse(e);
  }
  unwrap() {
    return this._def.innerType;
  }
}
ue.create = (t, e) => new ue({
  innerType: t,
  typeName: v.ZodNullable,
  ..._(e)
});
class Me extends k {
  _parse(e) {
    const { ctx: s } = this._processInputParams(e);
    let r = s.data;
    return s.parsedType === h.undefined && (r = this._def.defaultValue()), this._def.innerType._parse({
      data: r,
      path: s.path,
      parent: s
    });
  }
  removeDefault() {
    return this._def.innerType;
  }
}
Me.create = (t, e) => new Me({
  innerType: t,
  typeName: v.ZodDefault,
  defaultValue: typeof e.default == "function" ? e.default : () => e.default,
  ..._(e)
});
class ze extends k {
  _parse(e) {
    const { ctx: s } = this._processInputParams(e), r = {
      ...s,
      common: {
        ...s.common,
        issues: []
      }
    }, a = this._def.innerType._parse({
      data: r.data,
      path: r.path,
      parent: {
        ...r
      }
    });
    return be(a) ? a.then((n) => ({
      status: "valid",
      value: n.status === "valid" ? n.value : this._def.catchValue({
        get error() {
          return new q(r.common.issues);
        },
        input: r.data
      })
    })) : {
      status: "valid",
      value: a.status === "valid" ? a.value : this._def.catchValue({
        get error() {
          return new q(r.common.issues);
        },
        input: r.data
      })
    };
  }
  removeCatch() {
    return this._def.innerType;
  }
}
ze.create = (t, e) => new ze({
  innerType: t,
  typeName: v.ZodCatch,
  catchValue: typeof e.catch == "function" ? e.catch : () => e.catch,
  ..._(e)
});
class _t extends k {
  _parse(e) {
    if (this._getType(e) !== h.nan) {
      const r = this._getOrReturnCtx(e);
      return u(r, {
        code: c.invalid_type,
        expected: h.nan,
        received: r.parsedType
      }), y;
    }
    return { status: "valid", value: e.data };
  }
}
_t.create = (t) => new _t({
  typeName: v.ZodNaN,
  ..._(t)
});
class hr extends k {
  _parse(e) {
    const { ctx: s } = this._processInputParams(e), r = s.data;
    return this._def.type._parse({
      data: r,
      path: s.path,
      parent: s
    });
  }
  unwrap() {
    return this._def.type;
  }
}
class Ke extends k {
  _parse(e) {
    const { status: s, ctx: r } = this._processInputParams(e);
    if (r.common.async)
      return (async () => {
        const n = await this._def.in._parseAsync({
          data: r.data,
          path: r.path,
          parent: r
        });
        return n.status === "aborted" ? y : n.status === "dirty" ? (s.dirty(), fe(n.value)) : this._def.out._parseAsync({
          data: n.value,
          path: r.path,
          parent: r
        });
      })();
    {
      const a = this._def.in._parseSync({
        data: r.data,
        path: r.path,
        parent: r
      });
      return a.status === "aborted" ? y : a.status === "dirty" ? (s.dirty(), {
        status: "dirty",
        value: a.value
      }) : this._def.out._parseSync({
        data: a.value,
        path: r.path,
        parent: r
      });
    }
  }
  static create(e, s) {
    return new Ke({
      in: e,
      out: s,
      typeName: v.ZodPipeline
    });
  }
}
class Ve extends k {
  _parse(e) {
    const s = this._def.innerType._parse(e), r = (a) => (le(a) && (a.value = Object.freeze(a.value)), a);
    return be(s) ? s.then((a) => r(a)) : r(s);
  }
  unwrap() {
    return this._def.innerType;
  }
}
Ve.create = (t, e) => new Ve({
  innerType: t,
  typeName: v.ZodReadonly,
  ..._(e)
});
var v;
(function(t) {
  t.ZodString = "ZodString", t.ZodNumber = "ZodNumber", t.ZodNaN = "ZodNaN", t.ZodBigInt = "ZodBigInt", t.ZodBoolean = "ZodBoolean", t.ZodDate = "ZodDate", t.ZodSymbol = "ZodSymbol", t.ZodUndefined = "ZodUndefined", t.ZodNull = "ZodNull", t.ZodAny = "ZodAny", t.ZodUnknown = "ZodUnknown", t.ZodNever = "ZodNever", t.ZodVoid = "ZodVoid", t.ZodArray = "ZodArray", t.ZodObject = "ZodObject", t.ZodUnion = "ZodUnion", t.ZodDiscriminatedUnion = "ZodDiscriminatedUnion", t.ZodIntersection = "ZodIntersection", t.ZodTuple = "ZodTuple", t.ZodRecord = "ZodRecord", t.ZodMap = "ZodMap", t.ZodSet = "ZodSet", t.ZodFunction = "ZodFunction", t.ZodLazy = "ZodLazy", t.ZodLiteral = "ZodLiteral", t.ZodEnum = "ZodEnum", t.ZodEffects = "ZodEffects", t.ZodNativeEnum = "ZodNativeEnum", t.ZodOptional = "ZodOptional", t.ZodNullable = "ZodNullable", t.ZodDefault = "ZodDefault", t.ZodCatch = "ZodCatch", t.ZodPromise = "ZodPromise", t.ZodBranded = "ZodBranded", t.ZodPipeline = "ZodPipeline", t.ZodReadonly = "ZodReadonly";
})(v || (v = {}));
const mr = Q.create;
ee.create;
U.create;
const pr = S.create;
Te.create;
Ce.create;
ae.create;
de.create;
Ne.create;
G.create;
ue.create;
const gr = (t) => pr({
  password: mr().nonempty(t("required", { field: t("Password"), ns: "validation" }))
}), la = ({
  title: t,
  description: e,
  children: s,
  onSubmit: r,
  ...a
}) => {
  const { t: n } = bt(), i = ls(() => gr(n), [n]), o = Ts({
    resolver: Vs(i),
    defaultValues: {
      password: ""
    }
  });
  return /* @__PURE__ */ R.jsxs(qt, { ...a, children: [
    s && /* @__PURE__ */ R.jsx(Ht, { children: s }),
    /* @__PURE__ */ R.jsx(He, { children: /* @__PURE__ */ R.jsx(ks, { ...o, children: /* @__PURE__ */ R.jsxs("form", { onSubmit: o.handleSubmit(r), className: "space-y-3", children: [
      /* @__PURE__ */ R.jsxs(Xe, { children: [
        /* @__PURE__ */ R.jsx(Je, { children: t || n("Confirm your password") }),
        /* @__PURE__ */ R.jsxs("div", { className: "text-secondary text-body-sm-medium space-y-3", children: [
          e || n("Enter your account password to complete this action"),
          /* @__PURE__ */ R.jsx(
            bs,
            {
              required: !0,
              type: "password",
              control: o.control,
              name: "password",
              placeholder: n("Password")
            }
          )
        ] })
      ] }),
      /* @__PURE__ */ R.jsxs(Ye, { children: [
        /* @__PURE__ */ R.jsx(Ge, { type: "reset", onClick: () => o.reset(), children: n("Cancel") }),
        /* @__PURE__ */ R.jsx(Qe, { type: "submit", children: n("Confirm") })
      ] })
    ] }) }) })
  ] });
};
function es(t) {
  var e, s, r = "";
  if (typeof t == "string" || typeof t == "number") r += t;
  else if (typeof t == "object") if (Array.isArray(t)) {
    var a = t.length;
    for (e = 0; e < a; e++) t[e] && (s = es(t[e])) && (r && (r += " "), r += s);
  } else for (s in t) t[s] && (r && (r += " "), r += s);
  return r;
}
function re() {
  for (var t, e, s = 0, r = "", a = arguments.length; s < a; s++) (t = arguments[s]) && (e = es(t)) && (r && (r += " "), r += e);
  return r;
}
var ve = (t) => typeof t == "number" && !isNaN(t), ne = (t) => typeof t == "string", H = (t) => typeof t == "function", yr = (t) => ne(t) || ve(t), Be = (t) => ne(t) || H(t) ? t : null, vr = (t, e) => t === !1 || ve(t) && t > 0 ? t : e, Fe = (t) => Ae(t) || ne(t) || H(t) || ve(t);
function _r(t, e, s = 300) {
  let { scrollHeight: r, style: a } = t;
  requestAnimationFrame(() => {
    a.minHeight = "initial", a.height = r + "px", a.transition = `all ${s}ms`, requestAnimationFrame(() => {
      a.height = "0", a.padding = "0", a.margin = "0", setTimeout(e, s);
    });
  });
}
function xr({ enter: t, exit: e, appendPosition: s = !1, collapse: r = !0, collapseDuration: a = 300 }) {
  return function({ children: n, position: i, preventExitTransition: o, done: l, nodeRef: d, isIn: f, playToast: E }) {
    let Z = s ? `${t}--${i}` : t, L = s ? `${e}--${i}` : e, M = he(0);
    return kt(() => {
      let I = d.current, C = Z.split(" "), T = (p) => {
        p.target === d.current && (E(), I.removeEventListener("animationend", T), I.removeEventListener("animationcancel", T), M.current === 0 && p.type !== "animationcancel" && I.classList.remove(...C));
      };
      I.classList.add(...C), I.addEventListener("animationend", T), I.addEventListener("animationcancel", T);
    }, []), Ee(() => {
      let I = d.current, C = () => {
        I.removeEventListener("animationend", C), r ? _r(I, l, a) : l();
      };
      f || (o ? C() : (M.current = 1, I.className += ` ${L}`, I.addEventListener("animationend", C)));
    }, [f]), j.createElement(j.Fragment, null, n);
  };
}
function xt(t, e) {
  return { content: ts(t.content, t.props), containerId: t.props.containerId, id: t.props.toastId, theme: t.props.theme, type: t.props.type, data: t.props.data || {}, isLoading: t.props.isLoading, icon: t.props.icon, reason: t.removalReason, status: e };
}
function ts(t, e, s = !1) {
  return Ae(t) && !ne(t.type) ? qe(t, { closeToast: e.closeToast, toastProps: e, data: e.data, isPaused: s }) : H(t) ? t({ closeToast: e.closeToast, toastProps: e, data: e.data, isPaused: s }) : t;
}
function kr({ closeToast: t, theme: e, ariaLabel: s = "close" }) {
  return j.createElement("button", { className: `Toastify__close-button Toastify__close-button--${e}`, type: "button", onClick: (r) => {
    r.stopPropagation(), t(!0);
  }, "aria-label": s }, j.createElement("svg", { "aria-hidden": "true", viewBox: "0 0 14 16" }, j.createElement("path", { fillRule: "evenodd", d: "M7.71 8.23l3.75 3.75-1.48 1.48-3.75-3.75-3.75 3.75L1 11.98l3.75-3.75L1 4.48 2.48 3l3.75 3.75L9.98 3l1.48 1.48-3.75 3.75z" })));
}
function br({ delay: t, isRunning: e, closeToast: s, type: r = "default", hide: a, className: n, controlledProgress: i, progress: o, rtl: l, isIn: d, theme: f }) {
  let E = a || i && o === 0, Z = { animationDuration: `${t}ms`, animationPlayState: e ? "running" : "paused" };
  i && (Z.transform = `scaleX(${o})`);
  let L = re("Toastify__progress-bar", i ? "Toastify__progress-bar--controlled" : "Toastify__progress-bar--animated", `Toastify__progress-bar-theme--${f}`, `Toastify__progress-bar--${r}`, { "Toastify__progress-bar--rtl": l }), M = H(n) ? n({ rtl: l, type: r, defaultClassName: L }) : re(L, n), I = { [i && o >= 1 ? "onTransitionEnd" : "onAnimationEnd"]: i && o < 1 ? null : () => {
    d && s();
  } };
  return j.createElement("div", { className: "Toastify__progress-bar--wrp", "data-hidden": E }, j.createElement("div", { className: `Toastify__progress-bar--bg Toastify__progress-bar-theme--${f} Toastify__progress-bar--${r}` }), j.createElement("div", { role: "progressbar", "aria-hidden": E ? "true" : "false", "aria-label": "notification timer", className: M, style: Z, ...I }));
}
var wr = 1, ss = () => `${wr++}`;
function Tr(t, e, s) {
  let r = 1, a = 0, n = [], i = [], o = e, l = /* @__PURE__ */ new Map(), d = /* @__PURE__ */ new Set(), f = (p) => (d.add(p), () => d.delete(p)), E = () => {
    i = Array.from(l.values()), d.forEach((p) => p());
  }, Z = ({ containerId: p, toastId: g, updateId: x }) => {
    let $ = p ? p !== t : t !== 1, z = l.has(g) && x == null;
    return $ || z;
  }, L = (p, g) => {
    l.forEach((x) => {
      var $;
      (g == null || g === x.props.toastId) && (($ = x.toggle) == null || $.call(x, p));
    });
  }, M = (p) => {
    var g, x;
    (x = (g = p.props) == null ? void 0 : g.onClose) == null || x.call(g, p.removalReason), p.isActive = !1;
  }, I = (p) => {
    if (p == null) l.forEach(M);
    else {
      let g = l.get(p);
      g && M(g);
    }
    E();
  }, C = () => {
    a -= n.length, n = [];
  }, T = (p) => {
    var g, x;
    let { toastId: $, updateId: z } = p.props, w = z == null;
    p.staleId && l.delete(p.staleId), p.isActive = !0, l.set($, p), E(), s(xt(p, w ? "added" : "updated")), w && ((x = (g = p.props).onOpen) == null || x.call(g));
  };
  return { id: t, props: o, observe: f, toggle: L, removeToast: I, toasts: l, clearQueue: C, buildToast: (p, g) => {
    if (Z(g)) return;
    let { toastId: x, updateId: $, data: z, staleId: w, delay: A } = g, F = $ == null;
    F && a++;
    let Y = { ...o, style: o.toastStyle, key: r++, ...Object.fromEntries(Object.entries(g).filter(([se, xe]) => xe != null)), toastId: x, updateId: $, data: z, isIn: !1, className: Be(g.className || o.toastClassName), progressClassName: Be(g.progressClassName || o.progressClassName), autoClose: g.isLoading ? !1 : vr(g.autoClose, o.autoClose), closeToast(se) {
      l.get(x).removalReason = se, I(x);
    }, deleteToast() {
      let se = l.get(x);
      if (se != null) {
        if (s(xt(se, "removed")), l.delete(x), a--, a < 0 && (a = 0), n.length > 0) {
          T(n.shift());
          return;
        }
        E();
      }
    } };
    Y.closeButton = o.closeButton, g.closeButton === !1 || Fe(g.closeButton) ? Y.closeButton = g.closeButton : g.closeButton === !0 && (Y.closeButton = Fe(o.closeButton) ? o.closeButton : !0);
    let W = { content: p, props: Y, staleId: w };
    o.limit && o.limit > 0 && a > o.limit && F ? n.push(W) : ve(A) ? setTimeout(() => {
      T(W);
    }, A) : T(W);
  }, setProps(p) {
    o = p;
  }, setToggle: (p, g) => {
    let x = l.get(p);
    x && (x.toggle = g);
  }, isToastActive: (p) => {
    var g;
    return (g = l.get(p)) == null ? void 0 : g.isActive;
  }, getSnapshot: () => i };
}
var D = /* @__PURE__ */ new Map(), ye = [], Ue = /* @__PURE__ */ new Set(), Cr = (t) => Ue.forEach((e) => e(t)), rs = () => D.size > 0;
function Nr() {
  ye.forEach((t) => ns(t.content, t.options)), ye = [];
}
var Er = (t, { containerId: e }) => {
  var s;
  return (s = D.get(e || 1)) == null ? void 0 : s.toasts.get(t);
};
function as(t, e) {
  var s;
  if (e) return !!((s = D.get(e)) != null && s.isToastActive(t));
  let r = !1;
  return D.forEach((a) => {
    a.isToastActive(t) && (r = !0);
  }), r;
}
function Ar(t) {
  if (!rs()) {
    ye = ye.filter((e) => t != null && e.options.toastId !== t);
    return;
  }
  if (t == null || yr(t)) D.forEach((e) => {
    e.removeToast(t);
  });
  else if (t && ("containerId" in t || "id" in t)) {
    let e = D.get(t.containerId);
    e ? e.removeToast(t.id) : D.forEach((s) => {
      s.removeToast(t.id);
    });
  }
}
var Ir = (t = {}) => {
  D.forEach((e) => {
    e.props.limit && (!t.containerId || e.id === t.containerId) && e.clearQueue();
  });
};
function ns(t, e) {
  Fe(t) && (rs() || ye.push({ content: t, options: e }), D.forEach((s) => {
    s.buildToast(t, e);
  }));
}
function jr(t) {
  var e;
  (e = D.get(t.containerId || 1)) == null || e.setToggle(t.id, t.fn);
}
function is(t, e) {
  D.forEach((s) => {
    (e == null || !(e != null && e.containerId) || (e == null ? void 0 : e.containerId) === s.id) && s.toggle(t, e == null ? void 0 : e.id);
  });
}
function Or(t) {
  let e = t.containerId || 1;
  return { subscribe(s) {
    let r = Tr(e, t, Cr);
    D.set(e, r);
    let a = r.observe(s);
    return Nr(), () => {
      a(), D.delete(e);
    };
  }, setProps(s) {
    var r;
    (r = D.get(e)) == null || r.setProps(s);
  }, getSnapshot() {
    var s;
    return (s = D.get(e)) == null ? void 0 : s.getSnapshot();
  } };
}
function Rr(t) {
  return Ue.add(t), () => {
    Ue.delete(t);
  };
}
function Sr(t) {
  return t && (ne(t.toastId) || ve(t.toastId)) ? t.toastId : ss();
}
function _e(t, e) {
  return ns(t, e), e.toastId;
}
function Ie(t, e) {
  return { ...e, type: e && e.type || t, toastId: Sr(e) };
}
function je(t) {
  return (e, s) => _e(e, Ie(t, s));
}
function N(t, e) {
  return _e(t, Ie("default", e));
}
N.loading = (t, e) => _e(t, Ie("default", { isLoading: !0, autoClose: !1, closeOnClick: !1, closeButton: !1, draggable: !1, ...e }));
function $r(t, { pending: e, error: s, success: r }, a) {
  let n;
  e && (n = ne(e) ? N.loading(e, a) : N.loading(e.render, { ...a, ...e }));
  let i = { isLoading: null, autoClose: null, closeOnClick: null, closeButton: null, draggable: null }, o = (d, f, E) => {
    if (f == null) {
      N.dismiss(n);
      return;
    }
    let Z = { type: d, ...i, ...a, data: E }, L = ne(f) ? { render: f } : f;
    return n ? N.update(n, { ...Z, ...L }) : N(L.render, { ...Z, ...L }), E;
  }, l = H(t) ? t() : t;
  return l.then((d) => o("success", r, d)).catch((d) => o("error", s, d)), l;
}
N.promise = $r;
N.success = je("success");
N.info = je("info");
N.error = je("error");
N.warning = je("warning");
N.warn = N.warning;
N.dark = (t, e) => _e(t, Ie("default", { theme: "dark", ...e }));
function Zr(t) {
  Ar(t);
}
N.dismiss = Zr;
N.clearWaitingQueue = Ir;
N.isActive = as;
N.update = (t, e = {}) => {
  let s = Er(t, e);
  if (s) {
    let { props: r, content: a } = s, n = { delay: 100, ...r, ...e, toastId: e.toastId || t, updateId: ss() };
    n.toastId !== t && (n.staleId = t);
    let i = n.render || a;
    delete n.render, _e(i, n);
  }
};
N.done = (t) => {
  N.update(t, { progress: 1 });
};
N.onChange = Rr;
N.play = (t) => is(!0, t);
N.pause = (t) => is(!1, t);
function Lr(t) {
  var e;
  let { subscribe: s, getSnapshot: r, setProps: a } = he(Or(t)).current;
  a(t);
  let n = (e = ds(s, r, r)) == null ? void 0 : e.slice();
  function i(o) {
    if (!n) return [];
    let l = /* @__PURE__ */ new Map();
    return t.newestOnTop && n.reverse(), n.forEach((d) => {
      let { position: f } = d.props;
      l.has(f) || l.set(f, []), l.get(f).push(d);
    }), Array.from(l, (d) => o(d[0], d[1]));
  }
  return { getToastToRender: i, isToastActive: as, count: n == null ? void 0 : n.length };
}
function Pr(t) {
  let [e, s] = Ze(!1), [r, a] = Ze(!1), n = he(null), i = he({ start: 0, delta: 0, removalDistance: 0, canCloseOnClick: !0, canDrag: !1, didMove: !1 }).current, { autoClose: o, pauseOnHover: l, closeToast: d, onClick: f, closeOnClick: E } = t;
  jr({ id: t.toastId, containerId: t.containerId, fn: s }), Ee(() => {
    if (t.pauseOnFocusLoss) return Z(), () => {
      L();
    };
  }, [t.pauseOnFocusLoss]);
  function Z() {
    document.hasFocus() || T(), window.addEventListener("focus", C), window.addEventListener("blur", T);
  }
  function L() {
    window.removeEventListener("focus", C), window.removeEventListener("blur", T);
  }
  function M(w) {
    if (t.draggable === !0 || t.draggable === w.pointerType) {
      p();
      let A = n.current;
      i.canCloseOnClick = !0, i.canDrag = !0, A.style.transition = "none", t.draggableDirection === "x" ? (i.start = w.clientX, i.removalDistance = A.offsetWidth * (t.draggablePercent / 100)) : (i.start = w.clientY, i.removalDistance = A.offsetHeight * (t.draggablePercent === 80 ? t.draggablePercent * 1.5 : t.draggablePercent) / 100);
    }
  }
  function I(w) {
    let { top: A, bottom: F, left: Y, right: W } = n.current.getBoundingClientRect();
    w.nativeEvent.type !== "touchend" && t.pauseOnHover && w.clientX >= Y && w.clientX <= W && w.clientY >= A && w.clientY <= F ? T() : C();
  }
  function C() {
    s(!0);
  }
  function T() {
    s(!1);
  }
  function p() {
    i.didMove = !1, document.addEventListener("pointermove", x), document.addEventListener("pointerup", $);
  }
  function g() {
    document.removeEventListener("pointermove", x), document.removeEventListener("pointerup", $);
  }
  function x(w) {
    let A = n.current;
    if (i.canDrag && A) {
      i.didMove = !0, e && T(), t.draggableDirection === "x" ? i.delta = w.clientX - i.start : i.delta = w.clientY - i.start, i.start !== w.clientX && (i.canCloseOnClick = !1);
      let F = t.draggableDirection === "x" ? `${i.delta}px, var(--y)` : `0, calc(${i.delta}px + var(--y))`;
      A.style.transform = `translate3d(${F},0)`, A.style.opacity = `${1 - Math.abs(i.delta / i.removalDistance)}`;
    }
  }
  function $() {
    g();
    let w = n.current;
    if (i.canDrag && i.didMove && w) {
      if (i.canDrag = !1, Math.abs(i.delta) > i.removalDistance) {
        a(!0), t.closeToast(!0), t.collapseAll();
        return;
      }
      w.style.transition = "transform 0.2s, opacity 0.2s", w.style.removeProperty("transform"), w.style.removeProperty("opacity");
    }
  }
  let z = { onPointerDown: M, onPointerUp: I };
  return o && l && (z.onMouseEnter = T, t.stacked || (z.onMouseLeave = C)), E && (z.onClick = (w) => {
    f && f(w), i.canCloseOnClick && d(!0);
  }), { playToast: C, pauseToast: T, isRunning: e, preventExitTransition: r, toastRef: n, eventHandlers: z };
}
var Dr = typeof window < "u" ? kt : Ee, Oe = ({ theme: t, type: e, isLoading: s, ...r }) => j.createElement("svg", { viewBox: "0 0 24 24", width: "100%", height: "100%", fill: t === "colored" ? "currentColor" : `var(--toastify-icon-color-${e})`, ...r });
function Mr(t) {
  return j.createElement(Oe, { ...t }, j.createElement("path", { d: "M23.32 17.191L15.438 2.184C14.728.833 13.416 0 11.996 0c-1.42 0-2.733.833-3.443 2.184L.533 17.448a4.744 4.744 0 000 4.368C1.243 23.167 2.555 24 3.975 24h16.05C22.22 24 24 22.044 24 19.632c0-.904-.251-1.746-.68-2.44zm-9.622 1.46c0 1.033-.724 1.823-1.698 1.823s-1.698-.79-1.698-1.822v-.043c0-1.028.724-1.822 1.698-1.822s1.698.79 1.698 1.822v.043zm.039-12.285l-.84 8.06c-.057.581-.408.943-.897.943-.49 0-.84-.367-.896-.942l-.84-8.065c-.057-.624.25-1.095.779-1.095h1.91c.528.005.84.476.784 1.1z" }));
}
function zr(t) {
  return j.createElement(Oe, { ...t }, j.createElement("path", { d: "M12 0a12 12 0 1012 12A12.013 12.013 0 0012 0zm.25 5a1.5 1.5 0 11-1.5 1.5 1.5 1.5 0 011.5-1.5zm2.25 13.5h-4a1 1 0 010-2h.75a.25.25 0 00.25-.25v-4.5a.25.25 0 00-.25-.25h-.75a1 1 0 010-2h1a2 2 0 012 2v4.75a.25.25 0 00.25.25h.75a1 1 0 110 2z" }));
}
function Vr(t) {
  return j.createElement(Oe, { ...t }, j.createElement("path", { d: "M12 0a12 12 0 1012 12A12.014 12.014 0 0012 0zm6.927 8.2l-6.845 9.289a1.011 1.011 0 01-1.43.188l-4.888-3.908a1 1 0 111.25-1.562l4.076 3.261 6.227-8.451a1 1 0 111.61 1.183z" }));
}
function Br(t) {
  return j.createElement(Oe, { ...t }, j.createElement("path", { d: "M11.983 0a12.206 12.206 0 00-8.51 3.653A11.8 11.8 0 000 12.207 11.779 11.779 0 0011.8 24h.214A12.111 12.111 0 0024 11.791 11.766 11.766 0 0011.983 0zM10.5 16.542a1.476 1.476 0 011.449-1.53h.027a1.527 1.527 0 011.523 1.47 1.475 1.475 0 01-1.449 1.53h-.027a1.529 1.529 0 01-1.523-1.47zM11 12.5v-6a1 1 0 012 0v6a1 1 0 11-2 0z" }));
}
function Fr() {
  return j.createElement("div", { className: "Toastify__spinner" });
}
var We = { info: zr, warning: Mr, success: Vr, error: Br, spinner: Fr }, Ur = (t) => t in We;
function Wr({ theme: t, type: e, isLoading: s, icon: r }) {
  let a = null, n = { theme: t, type: e };
  return r === !1 || (H(r) ? a = r({ ...n, isLoading: s }) : Ae(r) ? a = qe(r, n) : s ? a = We.spinner() : Ur(e) && (a = We[e](n))), a;
}
var qr = (t) => {
  let { isRunning: e, preventExitTransition: s, toastRef: r, eventHandlers: a, playToast: n } = Pr(t), { closeButton: i, children: o, autoClose: l, onClick: d, type: f, hideProgressBar: E, closeToast: Z, transition: L, position: M, className: I, style: C, progressClassName: T, updateId: p, role: g, progress: x, rtl: $, toastId: z, deleteToast: w, isIn: A, isLoading: F, closeOnClick: Y, theme: W, ariaLabel: se } = t, xe = re("Toastify__toast", `Toastify__toast-theme--${W}`, `Toastify__toast--${f}`, { "Toastify__toast--rtl": $ }, { "Toastify__toast--close-on-click": Y }), os = H(I) ? I({ rtl: $, position: M, type: f, defaultClassName: xe }) : re(xe, I), et = Wr(t), tt = !!x || !l, Re = { closeToast: Z, type: f, theme: W }, ke = null;
  return i === !1 || (H(i) ? ke = i(Re) : Ae(i) ? ke = qe(i, Re) : ke = kr(Re)), j.createElement(L, { isIn: A, done: w, position: M, preventExitTransition: s, nodeRef: r, playToast: n }, j.createElement("div", { id: z, tabIndex: 0, onClick: d, "data-in": A, className: os, ...a, style: C, ref: r, ...A && { role: g, "aria-label": se } }, et != null && j.createElement("div", { className: re("Toastify__toast-icon", { "Toastify--animate-icon Toastify__zoom-enter": !F }) }, et), ts(o, t, !e), ke, !t.customProgressBar && j.createElement(br, { ...p && !tt ? { key: `p-${p}` } : {}, rtl: $, theme: W, delay: l, isRunning: e, isIn: A, closeToast: Z, hide: E, type: f, className: T, controlledProgress: tt, progress: x || 0 })));
}, Hr = (t, e = !1) => ({ enter: `Toastify--animate Toastify__${t}-enter`, exit: `Toastify--animate Toastify__${t}-exit`, appendPosition: e }), Xr = xr(Hr("bounce", !0)), Yr = { position: "top-right", transition: Xr, autoClose: 5e3, closeButton: !0, pauseOnHover: !0, pauseOnFocusLoss: !0, draggable: "touch", draggablePercent: 80, draggableDirection: "x", role: "alert", theme: "light", "aria-label": "Notifications Alt+T", hotKeys: (t) => t.altKey && t.code === "KeyT" };
function da(t) {
  let e = { ...Yr, ...t }, s = t.stacked, [r, a] = Ze(!0), n = he(null), { getToastToRender: i, isToastActive: o, count: l } = Lr(e), { className: d, style: f, rtl: E, containerId: Z, hotKeys: L } = e;
  function M(C) {
    let T = re("Toastify__toast-container", `Toastify__toast-container--${C}`, { "Toastify__toast-container--rtl": E });
    return H(d) ? d({ position: C, rtl: E, defaultClassName: T }) : re(T, Be(d));
  }
  function I() {
    s && (a(!0), N.play());
  }
  return Dr(() => {
    var C;
    if (s) {
      let T = n.current.querySelectorAll('[data-in="true"]'), p = 12, g = (C = e.position) == null ? void 0 : C.includes("top"), x = 0, $ = 0;
      Array.from(T).reverse().forEach((z, w) => {
        let A = z;
        A.classList.add("Toastify__toast--stacked"), w > 0 && (A.dataset.collapsed = `${r}`), A.dataset.pos || (A.dataset.pos = g ? "top" : "bot");
        let F = x * (r ? 0.2 : 1) + (r ? 0 : p * w);
        A.style.setProperty("--y", `${g ? F : F * -1}px`), A.style.setProperty("--g", `${p}`), A.style.setProperty("--s", `${1 - (r ? $ : 0)}`), x += A.offsetHeight, $ += 0.025;
      });
    }
  }, [r, l, s]), Ee(() => {
    function C(T) {
      var p;
      let g = n.current;
      L(T) && ((p = g.querySelector('[tabIndex="0"]')) == null || p.focus(), a(!1), N.pause()), T.key === "Escape" && (document.activeElement === g || g != null && g.contains(document.activeElement)) && (a(!0), N.play());
    }
    return document.addEventListener("keydown", C), () => {
      document.removeEventListener("keydown", C);
    };
  }, [L]), j.createElement("section", { ref: n, className: "Toastify", id: Z, onMouseEnter: () => {
    s && (a(!1), N.pause());
  }, onMouseLeave: I, "aria-live": "polite", "aria-atomic": "false", "aria-relevant": "additions text", "aria-label": e["aria-label"] }, i((C, T) => {
    let p = T.length ? { ...f } : { ...f, pointerEvents: "none" };
    return j.createElement("div", { tabIndex: -1, className: M(C), "data-stacked": s, style: p, key: `c-${C}` }, T.map(({ content: g, props: x }) => j.createElement(qr, { ...x, stacked: s, collapseAll: I, isIn: o(x.toastId, x.containerId), key: `t-${x.key}` }, g)));
  }));
}
export {
  oa as C,
  da as L,
  la as P,
  N as y
};
